{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "toc": true
   },
   "source": [
    "<h1>Table of Contents<span class=\"tocSkip\"></span></h1>\n",
    "<div class=\"toc\"><ul class=\"toc-item\"><li><span><a href=\"#Geting-ground-truth\" data-toc-modified-id=\"Geting-ground-truth-1\"><span class=\"toc-item-num\">1&nbsp;&nbsp;</span>Geting ground truth</a></span><ul class=\"toc-item\"><li><span><a href=\"#Sample-list-from-72vrt\" data-toc-modified-id=\"Sample-list-from-72vrt-1.1\"><span class=\"toc-item-num\">1.1&nbsp;&nbsp;</span>Sample list from 72vrt</a></span></li><li><span><a href=\"#List-of-samples-imported-to-GV\" data-toc-modified-id=\"List-of-samples-imported-to-GV-1.2\"><span class=\"toc-item-num\">1.2&nbsp;&nbsp;</span>List of samples imported to GV</a></span></li><li><span><a href=\"#Experiment-QC-Notes\" data-toc-modified-id=\"Experiment-QC-Notes-1.3\"><span class=\"toc-item-num\">1.3&nbsp;&nbsp;</span>Experiment QC Notes</a></span></li><li><span><a href=\"#Experiment-Comments\" data-toc-modified-id=\"Experiment-Comments-1.4\"><span class=\"toc-item-num\">1.4&nbsp;&nbsp;</span>Experiment Comments</a></span><ul class=\"toc-item\"><li><span><a href=\"#Reading-json-exported-from-Mongo\" data-toc-modified-id=\"Reading-json-exported-from-Mongo-1.4.1\"><span class=\"toc-item-num\">1.4.1&nbsp;&nbsp;</span>Reading json exported from Mongo</a></span></li></ul></li><li><span><a href=\"#Putting-it-all--together\" data-toc-modified-id=\"Putting-it-all--together-1.5\"><span class=\"toc-item-num\">1.5&nbsp;&nbsp;</span>Putting it all  together</a></span></li><li><span><a href=\"#Result\" data-toc-modified-id=\"Result-1.6\"><span class=\"toc-item-num\">1.6&nbsp;&nbsp;</span>Result</a></span></li><li><span><a href=\"#Getting-experiments-to-process\" data-toc-modified-id=\"Getting-experiments-to-process-1.7\"><span class=\"toc-item-num\">1.7&nbsp;&nbsp;</span>Getting experiments to process</a></span></li><li><span><a href=\"#Conclusion:\" data-toc-modified-id=\"Conclusion:-1.8\"><span class=\"toc-item-num\">1.8&nbsp;&nbsp;</span>Conclusion:</a></span></li></ul></li><li><span><a href=\"#Getting-the-ground-truth,-revisited:\" data-toc-modified-id=\"Getting-the-ground-truth,-revisited:-2\"><span class=\"toc-item-num\">2&nbsp;&nbsp;</span>Getting the ground truth, revisited:</a></span><ul class=\"toc-item\"><li><span><a href=\"#Get-the-experiments-to-be-processed:\" data-toc-modified-id=\"Get-the-experiments-to-be-processed:-2.1\"><span class=\"toc-item-num\">2.1&nbsp;&nbsp;</span>Get the experiments to be processed:</a></span></li></ul></li><li><span><a href=\"#Appendix\" data-toc-modified-id=\"Appendix-3\"><span class=\"toc-item-num\">3&nbsp;&nbsp;</span>Appendix</a></span><ul class=\"toc-item\"><li><span><a href=\"#Alternative-approaches-to-pull-stuff-from-MongoDB\" data-toc-modified-id=\"Alternative-approaches-to-pull-stuff-from-MongoDB-3.1\"><span class=\"toc-item-num\">3.1&nbsp;&nbsp;</span>Alternative approaches to pull stuff from MongoDB</a></span></li><li><span><a href=\"#Fixing-labels-that-were-badly-adjusted\" data-toc-modified-id=\"Fixing-labels-that-were-badly-adjusted-3.2\"><span class=\"toc-item-num\">3.2&nbsp;&nbsp;</span>Fixing labels that were badly adjusted</a></span></li></ul></li></ul></div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Geting ground truth\n",
    "\n",
    "This notebook presents the steps that were taken to extract training labels from data fetched MongoDB, MySQL and 72vrt."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Sample list from 72vrt\n",
    "\n",
    "Obtaining a list of sample names that were processed and produced TXTs from 72vrt.\n",
    "\n",
    "This runs into *argument list too long* error:\n",
    "\n",
    "```bash\n",
    "ls /biodata/results/NEB_P1/*/*/TXT/*.gz | sed 's/^\\///' | awk -F\"/\" -v OFS=\"\\t\" '{gsub(/\\.rma\\.gz$/,\"\",$7); print $5,$7}' > ~/all_good_samples.tsv\n",
    "```\n",
    "This works:\n",
    "```bash\n",
    "find /biodata/results/NEB_P1/*/*/TXT/ -name \"*.rma.gz\" | sed 's/^\\///' | awk -F\"/\" -v OFS=\"\\t\" '{gsub(/\\.rma\\.gz$/,\"\",$7); print $5,$7}' > ~/all_good_samples.tsv\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### List of samples imported to GV\n",
    "Obtaining list of all samples that were imported in GV (from 16vrt):\n",
    "\n",
    "```SQL\n",
    "mysql --user=root -p\n",
    "show databases;\n",
    "USE gv_biology;\n",
    "show tables;\n",
    "describe experiment;\n",
    "\n",
    "SELECT e.nbr, ch.caption\n",
    "INTO outfile '/tmp/all_gv_samples.tsv'\n",
    "FIELDS TERMINATED BY '\\t' LINES TERMINATED BY '\\n'\n",
    "FROM experiment e\n",
    "JOIN replica r ON e.id=r.experiment_id\n",
    "JOIN chip ch ON ch.replica_id=r.id;\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Experiment QC Notes\n",
    "\n",
    "Getting per experiment qc-notes (from 73vrt):\n",
    "\n",
    "```SQL\n",
    "mysql --user=root -p\n",
    "USE experiment_management;\n",
    "\n",
    "SELECT e.nbr, e.rejected, p.short_caption, CONCAT('\"',e.qc_notes,'\"')\n",
    "INTO outfile '/tmp/exp_qc_notes.tsv'\n",
    "FIELDS TERMINATED BY '\\t'\n",
    "LINES TERMINATED BY '\\n'\n",
    "FROM experiment e\n",
    "JOIN platform p ON p.id=e.platform_id;\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Experiment Comments\n",
    "Getting experiment-level comments (from 73-vrt).  \n",
    "[Mongo Docs - Aggregation Pipeline](https://docs.mongodb.com/manual/core/aggregation-pipeline/index.html)\n",
    "\n",
    "Connect and inspect:\n",
    "```SQL\n",
    "mongo\n",
    "show databases\n",
    "use experiment-management\n",
    "show collections\n",
    "db.experiments.findOne()\n",
    "\n",
    "db.experiments.findOne({}, {nbr:1, comments:1})\n",
    "db.experiments.findOne({}, {nbr:1, \"comments.comment\":1})\n",
    "```\n",
    "\n",
    "Fetch results to separate database:\n",
    "```SQL\n",
    "/* works, but the arrays are not completely concatenated */\n",
    "db.experiments.aggregate([\n",
    "  {_id: \"$nbr\", $project: {comments: \"$comments.comment\"}},\n",
    "  {$out: \"mhresults\"}\n",
    "])\n",
    "```\n",
    "\n",
    "Export to file (in bash):\n",
    "```bash\n",
    "mongoexport --db experiment-management -c mhresults --out /tmp/exp_comments.json\n",
    "```\n",
    "\n",
    "Drop the separate temporary db (in mongo):\n",
    "```SQL\n",
    "db.mhresults.drop()\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Reading json exported from Mongo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "data = []\n",
    "fpath=\"./labels/exp_comments.json\"\n",
    "for line in open(fpath, 'r'):\n",
    "    data.append(json.loads(line))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = dict()\n",
    "for el in data:\n",
    "    results[el[\"_id\"]] = \"; \".join(sum(el[\"comments\"], []))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('./labels/exp_comments_norm.json', 'w') as wf:\n",
    "    json.dump(results, wf, sort_keys = True, indent = 4)\n",
    "# done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Putting it all  together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%cd labels"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get all samples that could have been potentialy discarded due to QC issues (these samples processed but weere not imported to GV):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "comm -23 <(sort all_good_samples.tsv) <(sort all_gv_samples.tsv) > perhaps_badqc_samples.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get all samples that should have good qc and that we know of both on 72 and gv. Alternatively, could use the whole `all_gv_samples.tsv` instead."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "comm -12 <(sort all_good_samples.tsv) <(sort all_gv_samples.tsv) > processed_and_imported_samples.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Some samples are in GV but their data is not on 72. This is weird. For now, dont bother."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "comm -13 <(sort all_good_samples.tsv) <(sort all_gv_samples.tsv) > imported_but_not_processed_samples.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure out which patterns are indicatvie of bad qc (exp or samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\"\"\n",
      "\"113 F, 170 F and 206 F: bad QC. Rest of the samples were not clearly defined.\"\n",
      "\"13 samples (GSM595899, GSM595901, GSM595902, GSM595908, GSM595910, GSM595911, GSM596004, GSM596005, GSM596037, GSM596054, GSM596055, GSM596056, GSM596057) are duplicated in the study HS-02089 (GSE38718). Therefore only 15 unique samples from this study were annotated under the HS-02137.\"\n",
      "\"16 Bad NUSE and Correlation:\\\n",
      "\"17 samples excluded - analyzed by different platform (GPL8300)\"\n",
      "\"181 samples excluded based on duplicate detection, mostly with HS-00867 (see matrix file for more details), 2 samples excluded based on QC: GSM1176244, GSM1176218\"\n",
      "\"18 out of 18 samples are duplicates. Duplicate of exp. HS-01358: GSM920386_H2-Glom-HT1; GSM920394_H2-Glom-HT2;  GSM920396_H2-Glom-HT4; GSM920397_H2-Glom-HT5; GSM920398_H2-Glom-HT6;  GSM920399_H2-Glom-HT7; GSM920400_H2-Glom-HT8; GSM920387_H2-Glom-HT11; GSM920388_H2-Glom-HT12; GSM920389_H2-Glom-HT13; GSM920390_H2-Glom-HT14; GSM920391_H2-Glom-HT15; GSM920392_H2-Glom-HT16; GSM920393_H2-Glom-HT18. Duplicate of exp. HS-01094: GSM1146221_H2-Glom-TN996; GSM1146219_H2-Glom-TN2; GSM1146220_H2-Glom-TN4; GSM1146222_H2-Glom-TN997\"\n",
      "\"18 samples are duplicates; 4 samples new\\\n",
      "\"1 bad quality sample, GSM1404316_af1c703−22−J2−260706, was not included\"\n",
      "\"1 bad quality sample, GSM1754950_Wheat_72HAF_s2, was not included\"\n",
      "\"1 bad quality sample, GSM886240_RbF4-1_YF050429WHT06, was not included\"\n",
      "\"1 sample, GSM773415_B6R_4, was not included as it might be mislabeled\"\n",
      "\"1 sample was excluded due to lower quality: GSM1635336; and 5 samples need to be checked after import: GSM1635335; GSM1635349; GSM1635351, GSM1635353 and GSM1635306.\"\n",
      "\"20 Duplicates with HS-00730\\\n",
      "\"22 out of 23 samples are duplicates of HS-00881 samples.\"\n",
      "\"25 samples are duplicated in HS-01051 and/or HS-01359. Moreover samples GSM318470 and GSM318473 were excluded due to bad QC (high NUSE, RLE and low array-array correlation). Therefore only 25 samples were annotated in HS-02324 (GSE13682).\"\n",
      "\"25 samples are duplicated in HS-01239 (GSE36700). Therefore only 5 samples were annotated under HS-02416.\"\n",
      "\"25 samples excluded, because they are duplicates of HS-01318. Four samples excluded based on QC (see matrix file excluded sample IDs). Have re-run QC with the left samples. The original QC results are under HS-02642_old.\"\n",
      "\"27 arrays are already part of HS-00837\"\n",
      "\"28_1879.CEL excluded, RLE above 0.5, high NUSE (above 1.05), low array-array correlation.\\\n",
      "\"2 bad quality samples, GSM719193 and GSM719194, were not included\"\n",
      "\"2 samples, GSM896892_R108R_10 and GSM896891_R108N_1, seem to be swapped and were therefore not included\"\n",
      "\"2 samples with no RAW data: GSM275489, GSM275490; rest of samples OK\"\n",
      "\"30 out of 61 Arrays are already part of HS-00763\"\n",
      "\"34 samples were excluded from annotation because they were duplicates with samples which are in GV.\"\n",
      "\"4 CEL files (out of 21) were truncated and thus could not be included in GV\"\n",
      "\"59 samples were excluded from annotation because they were duplicates with samples which are in GV.\"\n",
      "\"5 samples are OK. One sample  GSM2107410 was excluded, there were problems with its processing in our pipeline.\"\n",
      "\"5 samples were excluded from annotation because they were duplicates with samples which are in GV.\"\n",
      "\"60 samples are duplicated in HS-01373 and/or HS-01376. Moreover samples GSM2460009, GSM2460028 and GSM2460063 were excluded due to bad QC (high NUSE, RLE and low array-array correlation). Therefore only 38 samples were annotated in HS-02323 (GSE93658).\"\n",
      "\"60 samples (GSM98144, GSM98145, GSM98148, GSM98149, GSM98152, GSM98153, GSM98155, GSM98158, GSM98159, GSM98161, GSM98163, GSM98166, GSM98167, GSM98169, GSM98171, GSM98174, GSM98175, GSM98177, GSM98180, GSM98182, GSM98184, GSM98186, GSM98187, GSM98190, GSM98192, GSM98195, GSM98204, GSM98209, GSM98210, GSM98211, GSM98212, GSM98213, GSM98214, GSM98215, GSM98216, GSM98217, GSM98218, GSM98219, GSM98220, GSM98221, GSM98222, GSM98223, GSM98224, GSM98225, GSM98226, GSM98227, GSM98228, GSM98230, GSM98234, GSM98235, GSM98236, GSM98239, GSM98240, GSM98241, GSM98243, GSM98244, GSM98245, GSM98251, GSM98254, GSM98258) were excluded because they are duplicates with samples from the study HS-02368. The rest of samples is with good QC.\"\n",
      "\"61 samples were excluded because of low QC.\"\n",
      "\"61 samples were excluded from annotation, because they were duplicates with samples which are in GV.\"\n",
      "\"67 samples (from 77) are duplicates, already  curated in the study HS-01599\"\n",
      "\"6 out of the 9 samples are part of another experiment , 3 samples are unique\"\n",
      "\"74 samples detected as duplicates with HS-00843, 2 samples excluded based on QC: GSM340353, GSM340230\"\n",
      "\"74 samples were excluded from annotation because they are duplicates with samples which are in GV.\"\n",
      "\"8 duplicates with RN-00109\"\n",
      "\"96/105 samples were found as duplicates in HS-00135 (GSE2603). GSE68892 study contains additional molecular typing of breast cancer samples, therefore the GSE number was added to the already existing HS-00135 together with all the new clinical characteristics.\"\n",
      "\"99 samples detected as duplicates with HS-00843, 2 samples excluded based on low array-array correlation (GSM512229, GSM340230)\"\n",
      "\"A frequent general pattern of the 8x60K agilent chip. The expression density plot shows two populations of probe sets, otherwise samples correlate well.\"\n",
      "\"All (especially the controls) have bad NUSE and correlation\"\n",
      "\"All have bad correlation; Bad QC: GSM1239301_0204_Canine_2_\"\n",
      "\"Alll samples are of good quality.\"\n",
      "\"all OK\"\n",
      "\"all OK; 3 Duplicates with HS-00060: GSM321621, GSM321622, GSM321623\"\n",
      "\"all OK, but it's not worth to annotate; the experiment consists only of 5 samples, one sample per each differentiation stage\"\n",
      "\"all part of HS-00559\"\n",
      "\"All sample are good quality.\"\n",
      "\"All sample are OK.\"\n",
      "\"All sample have a good quality.\"\n",
      "\"All samples a have a good quality.\"\n",
      "\"All samples annotated (based on discussion).\"\n",
      "\"All samples are already annotated as HS-01281\"\n",
      "\"All samples are duplicates of HS-01558.\"\n",
      "\"All samples are duplicates of samples in GSE18517\"\n",
      "\"All samples are duplicates (subserie) from HS-01045.\"\n",
      "\"All samples are duplicates with the sample of studies: HS-01094; HS-01358; HS-01361; HS-01364; and HS-01093.\"\n",
      "\"All samples are good quality/.\"\n",
      "\"All samples are good quality.\"\n",
      "\"All samples are good quality\"\n",
      "\"All samples are good quality. Samples GSM1171596, GSM1171598, GSM1171600, GSM1171602, GSM1171604 and GSM1171606 excluded from annotation due to Cy5 labeling.\"\n",
      "\"All samples are good quality. The platform is not yet supported; annotations are available as flat file.\"\n",
      "\"All samples are of good quality.\"\n",
      "\"All samples are of good quality. Based on gender checks, the gender of some samples had to be corrected and in patients, where gender was not matching between the meniscus and the articular cartilage samples, pateint ID was deleted.\"\n",
      "\"all samples are of good quality but are measured with dual labeling cy3/cy5\"\n",
      "\"All samples are of low quality\"\n",
      "\"all samples are ok\"\n",
      "\"all  samples are OK\"\n",
      "\"All samples are ok.\"\n",
      "\"All samples are ok\"\n",
      "\"All samples are OK.\"\n",
      "\"All samples are OK.\\\n",
      "\"All samples are OK\"\n",
      "\"All samples are OK.  GSE17671 is a subserie of  GSE17672.\"\n",
      "\"All samples are OK. In the matrix file there is a mistake in rep 4, rep 5, rep 6, rep 7, and rep 8 description in the group \"after weight loss\", all is shifted about one row. This mistake was corrected.\"\n",
      "\"All samples are OK. Only C3 labelled samples were annotated.\"\n",
      "\"All samples are OK. Originally, there were 54 samples in the matrix file, but QC was performed on 18 samples. Each sample was uploaded three times on the ArrayExpress (checked by Eva P.).\\\n",
      "\"All samples are OK. Study contains three other platforms (HS_AFFY_U133A; MM_AFFY_U74A_V2; MM_AFFY_430_2); and samples were annotated under HS-02257, MM-00726, and MM-00727.\"\n",
      "\"All samples are ok. The study publication is not available.\"\n",
      "\"All samples are with good qc.\"\n",
      "\"All samples are with good QC.\"\n",
      "\"All samples are with good qc. Superserie.\"\n",
      "\"All samples duplicates of Exp.: HS-00247, HS-00391, HS-00519, HS-01352, HS-01353\"\n",
      "\"All samples have a good qaulity.\"\n",
      "\"All sample shave a good quality.\"\n",
      "\"All samples  have a good quality.\"\n",
      "\"All samples have  a good quality.\"\n",
      "\"All samples have a  good quality.\"\n",
      "\"All samples have a good quality.\"\n",
      "\"All samples have a good quality\"\n",
      "\"All samples have a good quality. 15 duplicates (GSM1448900\\\n",
      "\"All samples have a good quality. Cy3/Cy5  - only Cy3 processed.\"\n",
      "\"All samples have a good quality. Duplicates found with the study HS-00087 (GSM19165, GSM19168, GSM20657, GSM20658, GSM20660, GSM20662). HS-00087 has not been curated. No sample exclusion occurred.\"\n",
      "\"All samples have a good quality.  GSE10082 was analyzed.\"\n",
      "\"All samples have a good quality. GSM264813 duplicate with GSM264806 and excluded\"\n",
      "\"All samples have a good quality. GSM338151, GSM338152, GSM338153, GSM338196, GSM338197, GSM338198, GSM338154, GSM338155, GSM338156, GSM338199, GSM338200, GSM338201 excluded, since they are duplicates.\"\n",
      "\"All samples have a good quality. In the matrix file, the samples´ description differs from the sample´s title. Based on the number of the animals used in each group (from the Materials and Method), sample title appears to to be correct.\"\n",
      "\"All samples have a good quality. Many samples excluded for duplicity. List available upon request.\"\n",
      "\"All samples have a good quality. Only Cy3 (vehicle treated samples) could be annotated. GV does not support Cy5 (BMP4 treated) labeling.\"\n",
      "\"All samples have  a good quality. Only GSM652330 will be checked on 41-vrt.\"\n",
      "\"All samples have a good quality. Original article has not been found, however study was annotated with information from an article, which seems to be related.\"\n",
      "\"All samples have a good quality. Part of superserie (GSE34463), where the second part was analyzed on a platform which is not supported in GV. The same set of samples is also part of another superserie (GSE68387).\"\n",
      "\"All samples have  a good quality. Sample GSM1321377 was excluded due to truncated file.\"\n",
      "\"All samples have a good quality. Sample GSM1859007 has a lower correlation with other samples.\"\n",
      "\"All samples have a good quality. The original article is not available.\"\n",
      "\"All samples have  a good quliaty.\"\n",
      "\"All samples have good qc.\"\n",
      "\"All samples have good QC.\"\n",
      "\"All samples have good qc. The platform is not yet supported; annotations are available as flat file.\"\n",
      "\"All samples have good qc. The sample GSM881029 has lower correlation in comparison with other samples, but the correlation coefficient is within acceptable range.\"\n",
      "\"All samples have  good quality.\"\n",
      "\"All samples have good quality. \\\n",
      "\"All samples have good quality.\"\n",
      "\"All samples have good quality. All samples are duplicates to MM-00569 (GSE27987). Study is rejected.\"\n",
      "\"All samples have good quality.  GSM842167_Ptc_2, GSM842168_Ptc_3, GSM842169_Ptc_4,.GSM842170_Ptc_5  have high NUSE but also a high correlation.\"\n",
      "\"All samples have good quality. Samples GSM337817, GSM337818, GSM337819, GSM337820 were excluded  based on the contamination with other cell type, not due low quality.\"\n",
      "\"All samples left although they differ in expression density plots.\"\n",
      "\"All samples of good quality.\"\n",
      "\"All samples of good quality\"\n",
      "\"All samples of good quality. Healthy control samples are duplicates of those used in HS-02340. Experiment merged with HS-2340 and entry HS-2341 set as obsolete.\"\n",
      "\"All samples of good quality. Monocyte samples have a much higher intesity than nurse-like cell samples and somewhat elevated NUSE values, but decided to keep them. Check correlation with other monocyte samples in GV after import.\"\n",
      "\"All samples of good quality. Two separate GSE entries merged under one experiment number, because they share the healthy control samples and come from the same team.\"\n",
      "\"all samples ok\"\n",
      "\"all samples OK.\"\n",
      "\"all samples OK\"\n",
      "\"All samples ok.\"\n",
      "\"All samples ok\"\n",
      "\"All samples Ok.\"\n",
      "\"All samples O.K.\"\n",
      "\"All samples OK.\"\n",
      "\"All samples OK\"\n",
      "\"all samples OK, better correlation in samples cultivated under the same conditions (monolayer culture 0,91 - 0,92; ALI culture (0,86 - 0,91); the same case as in the study HS-01796 (SubSeries GSE55458 RNA seq)\"\n",
      "\"All samples OK. Duplicates found in 2 sample from 6, duplicated samples have the same GSM number.  (MM-00565/CEL/GSM213051 vs. MM-00289/CEL/GSM213051, MM-00565/CEL/GSM213057) Mismatch in samples description in between studies!\"\n",
      "\"All samples OK. GSE50800 contains dual-labelled samples.It is not supported in GV.\"\n",
      "\"All samples ok. Nevertheless samples GSM501260 and GSM501261 have NUSE ~ 1.10 (borderline).\"\n",
      "\"All samples OK. Platform not available.\"\n",
      "\"All samples OK, some have RLE above 0.5.\"\n",
      "\"All samples OK. Strange raw expression KDE.\"\n",
      "\"All samples QC ok.\"\n",
      "\"All samples were by QCNM and DD analysis identified as duplicates (subserie) from the study HS-01045.\"\n",
      "\"all samples were detected as duplicates with HS-01358\"\n",
      "\"All samples were good quality.\"\n",
      "\"All samples were good quality. Experiment is on multiple platform and only data from GPL81 were curated.\"\n",
      "\"All samples were good quality. Experiment is on multiple platforms, but only data from GPL81 were curated\"\n",
      "\"All samples with good qc.\"\n",
      "\"All samples with good QC.\"\n",
      "\"All samples with good QC. The study publication is not available.\"\n",
      "\"All smaples ok\"\n",
      "\"all zip archives entered, but one chip was bad quality array\"\n",
      "\"almost all samples (56 out of 63)  are duplicates and already curated in the study HS-01599\"\n",
      "\"almost all samples identified as duplicates of COPD samples in several other studies\"\n",
      "\"Already annotated as a part of HS-02474\"\n",
      "\"already annotated as MM-00324\"\n",
      "\"already annotated as MM-00554\"\n",
      "\"annotated, but can't be imported, QCNM failed (no TXT files produced)\"\n",
      "\"Annotated in HS-02384.\"\n",
      "\"Annotation to be delivered as a flat file without QC (negative values issue).\"\n",
      "\"Array-Array Intensity Correlation plot (0.6-1), poor correlation among most of the samples around 0,65.\"\n",
      "\"available as flat file\"\n",
      "\"Available as flat file.\"\n",
      "\"Available as flat file\"\n",
      "\"Available as flat-file.\"\n",
      "\"Available as flat-file. Samples GSM1487492 and GSM1487496 were excluded because they are assigned as \" Human Reference RNA \" without any characteristics.\"\n",
      "\"Available as flat file. Study was requested by PMI, but curation was cancelled due to higher PMI priorities.\"\n",
      "\"bad array-array correlation: TW1677_u133plus2_061004.CEL\"\n",
      "\"bad correlation\"\n",
      "\"bad correlation and NUSE: GSM285732, GSM285738\"\n",
      "\"Bad correlation: GSM1085698, GSM1085701, GSM1085694\"\n",
      "\"Bad correlation: GSM1272595\"\n",
      "\"Bad Correlation: GSM340843; 11 Duplictaes with RN-00109\"\n",
      "\"Bad correlation: GSM646498, GSM646499\"\n",
      "\"Bad Correlation: GSM709433_251485039461_1_1, GSM709452_251485039465_1_4\"\n",
      "\"Bad correlation: GSM788307_A662_05 and GSM788318_A662_16\"\n",
      "\"Bad Correlation: GSM807906\"\n",
      "\"Bad correlation (NUSE): GSM1064266, GSM1064267, GSM1064268, GSM1064269\"\n",
      "\"Bad correlation (NUSE): GSM231253\"\n",
      "\"Bad correlation (NUSE, low array-array correl.): GSM1318201, GSM1318203, GSM1318209, GSM1318215, GSM13182019\"\n",
      "\"Bad Correlation (whole experiment RLE processed looks weird): GSM636943_9−12−07−20812−PMT450_53.tiff_GE1−v5_95_Feb07_1_1, GSM636986_251485027273_GE1−v5_95_Feb07_1_3, GSM637026_B_251485027241_S01_GE1−v5_95_Feb07_1_2, GSM637056_B_251485027545_S01_GE1−v5_95_Feb07_1_4, GSM637199_B_251485027405_S01_GE1−v5_95_Feb07_1_4, GSM637236_B_251485027265_S01_GE1−v5_95_Feb07_1_1, GSM637313_B_251485027499_S01_GE1−v5_95_Feb07_1_2\"\n",
      "\"Bad Nuse:\\\n",
      "\"bad NUSE and array-array correlation: GSM690704; added SV:”tumor-infiltrating cell”; 10.10.2016 KH\"\n",
      "\"Bad NUSE and Correlation: GSM1041382, GSM1041384\"\n",
      "\"Bad NUSE and correlation: GSM1067814_Dog_12\"\n",
      "\"Bad NUSE and Correlation: GSM1181529_INT34, GSM1181602_INT101, GSM1181612_INT111, GSM1181624_INT123, GSM1181656_INT148, GSM1181670_INT160, GSM1181672_INT162, GSM1181673_INT163, GSM1181713_INT203, GSM1181718_INT208, GSM1181722_INT212, GSM1181748_INT238, GSM1181800_INT290\"\n",
      "\"bad NUSE and correlation: GSM1226172\"\n",
      "\"Bad NUSE and Correlation: GSM1296669_107−K, GSM1296670_108−K, GSM1296687_BxBC_30K_22mo, GSM1296688_BxBC_31K_77mo, GSM1296697_98K1mo, GSM1296701_BxBC_11K_32mo, GSM1296702_BxBC_16K_3mo, GSM1296705_27K9mo, GSM1296706_31K9mo, GSM1296707_37K9mo\"\n",
      "\"Bad NUSE and Correlation: GSM1329362_PF−1−MAHCC3−1R35CS−1−U133_2_2_madbID99452\"\n",
      "\"Bad NUSE and Correlation: GSM1414858_EA1−HuGene−1_0−ST−9967 / 18 Duplicates with HS-01373: GSM1414837_EA1-HuGene-1_0-ST-9522, GSM1414839_EA1-HuGene-1_0-ST-9524, GSM1414840_EA1-HuGene-1_0-ST-10462, GSM1414841_EA1-HuGene-1_0-ST-9966, GSM1414842_EA1-HuGene-1_0-ST-9968, GSM1414847_EA1-HuGene-1_0-ST-9570, GSM1414848_EA1-HuGene-1_0-ST-9625, GSM1414849_EA1-HuGene-1_0-ST-9523, GSM1414851_EA1-HuGene-1_0-ST-9567, GSM1414852_EA1-HuGene-1_0-ST-9568, GSM1414853_EA1-HuGene-1_0-ST-9569, GSM1414854_EA1-HuGene-1_0-ST-9526, GSM1414855_EA1-HuGene-1_0-ST-9626, GSM1414856_EA1-HuGene-1_0-ST-9573, GSM1414859_EA1-HuGene-1_0-ST-9969, GSM1414860_EA1-HuGene-1_0-ST-9971, GSM1414861_EA1-HuGene-1_0-ST-10811, GSM1414862_EA1-HuGene-1_0-ST-10814. Bad QC: GSM1414858_EA1-HuGene-1_0-ST-9967\"\n",
      "\"Bad NUSE and Correlation: GSM1523414_JZN_0059T_U133_2\"\n",
      "\"Bad NUSE and Correlation: GSM1536610_197RDB0414, 18 Duplicates with HS-00602 (of which 4 have a bad QC as well)\"\n",
      "\"Bad NUSE and correlation: GSM1550552_Increased_ALP_K9OSATissue3, GSM1550555_Increased_ALP_K9OSATissue6\"\n",
      "\"Bad NUSE and Correlation: GSM1559361_T314\"\n",
      "\"Bad NUSE and Correlation: GSM1623315__17, GSM1623352__101, GSM1623356__105\"\n",
      "\"Bad NUSE and correlation: GSM1911069_Dog5_500ug_24hr_Canine_2_\"\n",
      "\"Bad Nuse and Correlation: GSM370976, GSM370984\"\n",
      "\"Bad NUSE and correlation: GSM401533\"\n",
      "\"Bad NUSE and correlation: GSM429071\"\n",
      "\"Bad NUSE and Correlation: GSM498452_UZHRCC008_HG-U133, GSM498454_UZHRCC010_HG-U133, GSM498455_UZHRCC011_HG-U133, GSM498458_UZHRCC014_HG-U133, GSM498467_UZHRCC023_HG-U133, GSM498469_UZHRCC025_HG-U133, GSM498471_UZHRCC027_HG-U133, GSM498504_UZHRCC060_HG-U133, GSM498527_UZHRCC083_HG-U133, GSM498529_UZHRCC085_HG-U133, GSM498530_UZHRCC086_HG-U133, GSM498533_UZHRCC089_HG-U133, GSM498534_UZHRCC090_HG-U133, GSM498538_UZHRCC094_HG-U133, GSM498541_UZHRCC097_HG-U133, GSM498569_UZHRCC125_HG-U133 NO QC: GSM498570_UZHRCC126_HG-U133, GSM498571_UZHRCC127_HG-U133, GSM498572_UZHRCC128_HG-U133, GSM498573_UZHRCC129_HG-U133, GSM498574_UZHRCC130_HG-U133, GSM498575_UZHRCC131_HG-U133, GSM498576_UZHRCC132_HG-U133, GSM498577_UZHRCC133_HG-U133, GSM498578_UZHRCC134_HG-U133, GSM498579_UZHRCC135_HG-U133, GSM498580_UZHRCC136_HG-U133, GSM498581_UZHRCC137_HG-U133, GSM498582_UZHRCC138_HG-U133, GSM498583_UZHRCC139_HG-U133, GSM498584_UZHRCC140_HG-U133, GSM498585_UZHRCC141_HG-U133, GSM498586_UZHRCC142_HG-U133, GSM498587_UZHRCC143_HG-U133, GSM498588_UZHRCC144_HG-U133, GSM498589_UZHRCC145_HG-U133, GSM498590_UZHRCC146_HG-U133, GSM498591_UZHRCC147_HG-U133, GSM498592_UZHRCC148_HG-U133, GSM498593_UZHRCC149_HG-U133, GSM498594_UZHRCC150_HG-U133, GSM498595_UZHRCC151_HG-U133, GSM498596_UZHRCC152_HG-U133\"\n",
      "\"Bad NUSE and correlation: GSM502599, GSM502601, GSM502602, GSM502603\"\n",
      "\"Bad NUSE and Correlation: GSM557865, GSM557900, GSM557915\"\n",
      "\"bad NUSE and correlation: GSM559689\"\n",
      "\"Bad NUSE and Correlation: GSM562647_LNTOL12; 11 duplicates with HS-00476 and HS-00665\"\n",
      "\"Bad NUSE and correlation:  GSM609631\"\n",
      "\"Bad NUSE and Correlation: GSM714254, GSM714264, GSM714275, GSM714276 Duplicates with HS-01459: GSM714239, GSM714265, GSM714266, GSM714270, GSM714277, GSM714278, GSM714280, GSM714281, GSM714285, GSM714287, GSM714288\"\n",
      "\"Bad NUSE and Correlation: GSM714867, GSM714893, GSM714896, GSM714905, GSM715054, GSM715080, GSM715113, GSM715201, GSM715300 Duplicates with HS-01057: GSM714954 , GSM714955 , GSM714956 , GSM714960 , GSM714961 , GSM714962 , GSM714963 , GSM714964 , GSM714965 , GSM714969 , GSM714990 , GSM714991 , GSM714992 , GSM714996 , GSM714997 , GSM714998 , GSM715005 , GSM715006 , GSM715007 , GSM715014 , GSM715015 , GSM715016 , GSM715017 , GSM715018 , GSM715019 , GSM715080 , GSM715081 , GSM715082 , GSM715134 , GSM715140 , GSM715141 , GSM715142 , GSM715144 , GSM715145 , GSM715146 , GSM715147 , GSM715148 , GSM715149 , GSM715150 , GSM715151 , GSM715152\"\n",
      "\"Bad Nuse and Correlation: GSM920052_H7−Tub−LD1164\"\n",
      "\"Bad NUSE and Correlation: GSM955702_DNA10204−049, GSM955714_DNA10204−073, GSM955725_DNA10204−095, GSM955777_DNA10204−343, GSM955803_DNA11091−045, GSM955807_DNA11091−061\"\n",
      "\"Bad NUSE and Correlation: GSM992315, GSM992320, GSM992287\"\n",
      "\"Bad NUSE: GSM1134780_Par075HG−U133_Plus_2, GSM1134818_Par113HG−U133_Plus_2, GSM1134824_Par119HG−U133_Plus_2, GSM1134880_Par045HG−U133_Plus_2, GSM1134881_Par046HG−U133_Plus_2, GSM1134882_Par047HG−U133_Plus_2, GSM1134883_Par049HG−U133_Plus_2, GSM1304420_Par007U133Plus2.0, GSM1304421_Par008U133Plus2.0, GSM1304422_Par041U133Plus2.0, GSM1304423_Par042U133Plus2.0\"\n",
      "\"Bad NUSE: GSM1143810_ALL_294, GSM1143819_ALL_369, GSM1143859_ALL_552, GSM1143867_ALL_1\"\n",
      "\"Bad NUSE: GSM1300065_Cop051HG−U133_Plus_2, GSM1300092_Cop078HG−U133_Plus_2, GSM1300149_Cop139HG−U133_Plus_2, GSM1300153_Cop143HG−U133_Plus_2, GSM1300197_Cop187HG−U133_Plus_2\"\n",
      "\"Bad NUSE: GSM1339625_SynM−R57.CEL.gz\"\n",
      "\"Bad NUSE: GSM1404366, GSM1404367, GSM1404368\"\n",
      "\"Bad NUSE: GSM1653828_10_D0, GSM1653832_11_D8, GSM1653861_20_D8\"\n",
      "\"Bad NUSE: GSM204002\"\n",
      "\"Bad NUSE: GSM336980, GSM336983, GSM337004, GSM337011, GSM337010, GSM337029, GSM337065, GSM337066, GSM337069\"\n",
      "\"Bad NUSE: GSM338676, GSM338690, GSM338713, GSM338718, GSM338720, GSM338730, GSM338727, GSM338739, GSM338759, GSM338776, GSM338782, GSM338826 Unclear tumor-subtype, not annotated: GSM338790,GSM338799,GSM338800, GSM338801, GSM338802\"\n",
      "\"BAD NUSE: GSM538765, GSM538766\"\n",
      "\"Bad NUSE: GSM559752\"\n",
      "\"Bad NUSE: GSM573553, GSM573557,GSM573558, GSM573559, GSM573567\"\n",
      "\"Bad NUSE: GSM672494, GSM672495, GSM672496, GSM672497, GSM672499\\\n",
      "\"Bad Nuse: GSM920404_H4-Glom-IgA13, GSM920411_H4-Glom-IgA1\"\n",
      "\"Bad NUSE: GSM959890_AD11_Lesional_Pla168_Canine_2\"\n",
      "\"Bad NUSE: GSM966640_AMC25_Canine_2, GSM966641_AMC36_Canine_2\\\n",
      "\"bad QC: 071038-700\"\n",
      "\"bad QC: 164.txt\"\n",
      "\"bad QC: A568b_03_A2528.CEL, A568_C2425.CEL, A523_A2213.CEL, A523_C2195.CEL, A523_E2222.CEL\"\n",
      "\"Bad QC and Correlation: GSM1104671_FBM−5_S1, GSM1104679_FBM−6_S2, GSM1104694_FBM−8_S4\"\n",
      "\"Bad QC and Correlation: GSM414320\"\n",
      "\"Bad QC: BATCH1_RNA-R-C-C10.CEL.gz, BATCH1_RNA-R-B-G6.CEL.gz, BATCH2_RNA-R-A-G5.CEL.gz, BATCH3_RNA-R-B-B6.CEL.gz, BATCH1_RNA-R-C-E7.CEL.gz, BATCH1_RNA-R-C-G11.CEL.gz\"\n",
      "\"Bad QC (correlation):\\\n",
      "\"Bad QC (Correlation):\\\n",
      "\"Bad QC: correlation and Muse: Immcon2cd14poscd16neg241104.CEL.gz\"\n",
      "\"Bad QC (Correlation and Nuse):\\\n",
      "\"BAD QC (Correlation and NUSE): whole experiment not nice.\\\n",
      "\"Bad QC, do not annotate. Low array-array correlation (0.6) and internal duplicates\"\n",
      "\"Bad QC: GSM1003064, GSM1003065, GSM1003066, GSM1003073\"\n",
      "\"bad QC: GSM1020622\"\n",
      "\"bad QC: GSM1035944, GSM1035948\"\n",
      "\"Bad QC:GSM1193320, GSM1193343, GSM1193359, GSM1193380, GSM1193382, GSM1193383\"\n",
      "\"Bad QC: GSM1213886 GSM1213904 GSM1213906 GSM1213908 GSM1213927\"\n",
      "\"Bad QC: GSM1249377; merged to RM-00014\"\n",
      "\"bad QC: GSM1257481, GSM1257482, GSM1257483, GSM1257484; unclear annotation: GSM1257466, GSM1257467, GSM1257468, GSM1257469, GSM1257470, GSM1257471, GSM1257472, GSM1257473, GSM1257474, GSM1257475, GSM1257476, GSM1257477, GSM1257478, GSM1257479, GSM1257480\"\n",
      "\"bad QC: GSM1287965_dkfz1247, GSM1287968_dkfz1251\"\n",
      "\"bad QC: GSM1298500, GSM1298515, GSM1298609, GSM1298509, GSM1298607, GSM1298443, GSM1298574\"\n",
      "\"Bad QC: GSM1328247, GSM1328248, GSM1328249\"\n",
      "\"Bad QC: GSM1358964_H300_HG-U133_Plus_2_\"\n",
      "\"Bad QC: GSM1413940_59_HGU133_plus_2.0_Susan_04112013, GSM1413948_71_HGU133_plus_2.0_Susan_04112013, GSM1413946_67_HGU133_plus_2.0_Susan_04112013, GSM1413949_72_HGU133_plus_2.0_Susan_04112013, GSM1413947_68_HGU133_plus_2.0_Susan_04112013\"\n",
      "\"Bad QC: GSM1432970, GSM1432971\"\n",
      "\"Bad QC: GSM1513662\"\n",
      "\"Bad QC: GSM1700904\"\n",
      "\"Bad QC: GSM240809\"\n",
      "\"Bad QC: GSM244499, GSM201266. Replicates with HS-00391: GSM244488,GSM244489,GSM244490,GSM244491,GSM244492,GSM244493,GSM244494,GSM244495,GSM244496,GSM244497,GSM244498,GSM244507,GSM244508,GSM244509,GSM244510 --> All Con samples...therefore exp. was merged.\"\n",
      "\"bad QC: GSM257763, GSM257758\"\n",
      "\"Bad QC: GSM286412\"\n",
      "\"bad QC: GSM342613.CEL.gz,GSM342675.CEL.gz, GSM342857.CEL.gz, GSM342673.CEL.gz, GSM342844.CEL.gz, GSM342850.CEL.gz, GSM342884.CEL.gz, GSM342900.CEL.gz, GSM342814.CEL.gz, GSM342808.CEL.gz, GSM342879.CEL.gz, GSM342836.CEL.gz, GSM342737.CEL.gz, GSM342954.CEL.gz; another 40 samples excluded because CEL files were missing.\"\n",
      "\"bad QC: GSM344572\"\n",
      "\"Bad QC: GSM350290 GSM350320 GSM350359\"\n",
      "\"bad QC: GSM359407\"\n",
      "\"Bad QC: GSM362117\"\n",
      "\"Bad QC: GSM382283 GSM382306\"\n",
      "\"bad QC: GSM384876\"\n",
      "\"bad QC: GSM391601\"\n",
      "\"bad QC: GSM405522\"\n",
      "\"Bad QC: GSM405538\"\n",
      "\"bad QC: GSM410854, GSM411655, GSM411656, GSM411131, GSM410946\"\n",
      "\"Bad QC: GSM488188\"\n",
      "\"Bad QC: GSM493255.CEL.gz\"\n",
      "\"bad QC: GSM540672 and GSM540678\"\n",
      "\"bad QC: GSM597260, GSM597261\"\n",
      "\"Bad QC: GSM605037; added SV:”tumor-infiltrating cell”; 10.10.2016 KH\"\n",
      "\"bad QC: GSM626405, GSM626422, GSM626433\"\n",
      "\"bad QC: GSM636986, GSM637288, GSM637037, GSM637334, GSM636961, GSM636978, GSM637314, GSM636944, GSM636945, GSM636993, GSM636970, GSM636997, GSM636943, GSM637192, GSM637366, GSM636990, GSM636955, GSM637344, GSM637296, GSM637348, GSM637253, GSM637120, GSM637345, GSM637341, GSM636958, GSM636956, GSM637217, GSM637199, GSM637026, GSM637056, GSM637236, GSM637300, GSM637316, GSM637290, GSM637313, GSM637216, GSM637297\"\n",
      "\"Bad QC: GSM662176\"\n",
      "\"Bad QC: GSM705323_mRNA_mDC_EXP16, GSM705324_mRNA_mDC_EXP17, GSM705325_mRNA_mDC_EXP18, GSM705321_mRNA_mDC_EXP14, GSM705298_mRNA_CD19_EXP6, GSM705309_mRNA_CD56_EXP6, GSM705315_mRNA_CD8_EXP3, GSM705407_CD14_05_08_10,\"\n",
      "\"Bad QC: GSM761742\"\n",
      "\"bad QC: GSM773272\"\n",
      "\"bad QC: GSM790829_251485022507_1_2.txt.gz; GSM790822_251485022505_1_3.txt.gz\"\n",
      "\"Bad QC: GSM791076\"\n",
      "\"bad QC: GSM809172, GSM809175, GSM809178, GSM809181\"\n",
      "\"bad QC: GSM822725\"\n",
      "\"Bad QC: GSM846900, GSM846901\"\n",
      "\"bad QC: GSM866360_p0813_5nov\"\n",
      "\"bad QC: GSM877407, GSM877408\"\n",
      "\"bad QC: GSM899026, GSM899018\"\n",
      "\"Bad QC: GSM915832 (superserie of GSE37311 and GSE37312)\"\n",
      "\"bad QCNM score: GSM1511438\"\n",
      "\"bad QCNM score: GSM1585734, GSM1585747, GSM1585755, GSM1585762\"\n",
      "\"bad QCNM score: GSM723850, GSM723852\"\n",
      "\"Bad QC (NUSE) GSE68735: GSM1679902_9906_P7E03, GSM1679919_9906_P3E03, GSM1679944_9906_P3B09, GSM1679947_9906_P9A01, GSM1679977_9906_P2D12, GSM1679980_9906_P1A10, GSM1679988_9906_P1F08, GSM1679989_9906_P1B02, GSM1679991_9906_P4A03, GSM1680015_9906_P2F03, GSM1680021_9906_P7G04, GSM1680025_9906_P7C01, GSM1680029_9906_P7H04, GSM1680044_9906_P9C02, GSM1680047_9906_P1E03, GSM1680070_9906_P2E03, GSM1680074_9906_P1G10, GSM1680075_9906_P1G07, GSM1680078_9906_P1A09, GSM1680080_9906_P9C01, GSM1680092_9906_P3F03 Bad QC (NUSE) GSE11877: GSM299867, GSM299868, GSM299871, GSM299893, GSM299903, GSM299908, GSM299909, GSM299956, GSM299967, GSM299979, GSM299984, GSM299999, GSM300039, GSM300040, GSM300041, GSM300042, GSM300052, GSM300053, GSM300062\"\n",
      "\"Bad QC: P15001_920442_GRT12_05391.CEL.gz P15001_921157_GRT12_05394.CEL.gz P15001_921006_GRT12_05616.CEL.gz P15001_920908_GRT12_00231.CEL.gz P15001_921073_921073.CEL.gz P15001_921254_GRT13_01250_01.CEL.gz P15001_920838_GRT13_01362_01.CEL.gz P15001_920931_GRT13_01391_01.CEL.gz P15001_920125_GRT13_01207_01.CEL.gz P15001_920254_GRT13_01308_01.CEL.gz P15001_920410_GRT13_01325_01.CEL.gz P15001_GRT13_01383_920385_COPDII_Blood_7M_11122013.CEL.gz P15001_GRT13_01268_920523_COPDII_Blood_7M_11122013.CEL.gz P15001_920163_GRT13_01357_01.CEL.gz P15001_920370_GRT13_01368_01.CEL.gz P15001_920956_920956.CEL.gz P15001_GRT13_01228_920650_COPDII_Blood_7M_11122013.CEL.gz P15001_920932_GRT13_01410_01.CEL.gz P15001_920586_GRT13_01345_01.CEL.gz P15001_920697_GRT13_01248_01.CEL.gz P15001_920940_GRT13_01355_01.CEL.gz P15001_921251_GRT13_01331_01.CEL.gz P15001_920420_GRT13_01350_01.CEL.gz P15001_GRT13_01320_920001_COPDII_Blood_7M_11122013.CEL.gz P15001_920830_GRT13_01216_01.CEL.gz P15001_920858_GRT13_01403_01.CEL.gz P15001_921032_GRT13_01397_01.CEL.gz P15001_GRT13_01340_920267_COPDII_Blood_7M_11122013.CEL.gz\"\n",
      "\"bad QC: TwinFat119708AnAu50.1LeaPuhakka.CEL, TwinFat113201TaKu12.1LeaPuhakka.CEL, TwinFat128810MaTa21.1LeaPuhakka.CEL, TwinFat113202KaKu26.2LeaPuhakka.CEL, 128811KaSa30.1LeaPuhakka.CEL\"\n",
      "\"bad QC - very low correlation between samples\"\n",
      "\"Bad quality\"\n",
      "\"Bad quality of all samples\"\n",
      "\"Bad quality samples, GSM1563119_Wheat_seed_H2O2em-1 and all the endosperm samples, were not included\"\n",
      "\"BS_ni (MQ_44), MPG_ecz, MPG_pso: bad QC; duplicated experiment with HS-01107\"\n",
      "\"CEL file corrupded: GSM1060621_MC_SH_1-demi_2_HuGene-1_0-st-v1_\"\n",
      "\"check if all data are on the same platform\\\n",
      "\"chio Nr, GSM25798_JB−n_0821.cel.gz bad quality? not yet deleted\"\n",
      "\"chip Nr. GSM1141237_2060_4367_24173_168.0_HG−U133+Plus+2 entfernt\"\n",
      "\"chip Nr.: GSM1150042_LT123313RM_CTRL, GSM1150016_LT100984RL_COPD deleted due bad QC\"\n",
      "\"chip Nr. GSM518757_US45102978_251485018283_S01_GE1-v5_95_Feb07_1_2,\"\n",
      "\"chip Nr. GSM855911_Sample_2 deleted due to bad QC\"\n",
      "\"control neuron samples have high RLE, but otherwise look OK , therefore the samples were included into GV.\"\n",
      "\"Control samples GSM120065 and GSM120066 excluded from annotation because higher intensity in NUSE graph and low intensity correlation.\"\n",
      "\"Curation finished, waiting for QC (Martin Holub).\"\n",
      "\"curation in progress / Dual Dye\"\n",
      "\"currently not assigned to a correct platform\"\n",
      "\"Cy3/Cy5 double labeling, can't be integrated into GV.\"\n",
      "\"Cy3/Cy5 labeling - rejected.\"\n",
      "\"Cy3/Cy5 labeling -study rejected\"\n",
      "\"Cy3/Cy5 labeling used, in case of only Cy3 samples annotation, the design would be damaged. All Cy3 samples are OK.\"\n",
      "\"Cy5. Cannot process. To be delivered as a flat file without QC.\"\n",
      "\"Cy5/Cy3 labeling, not supported by GV.\"\n",
      "\"Data are measured in Cy5 channel.\"\n",
      "\"Data only for 4 samples were downloaded, samples are poorly and not clearly described.\"\n",
      "\"Deleted due to Duplicate (HS-00888)\"\n",
      "\"Deleted due to Duplicate (HS-00979)\"\n",
      "\"deleted due to quality problems\"\n",
      "\"Deleted due to quality problems\"\n",
      "\"deleted from GV due to low quality\"\n",
      "\"do not annotate, not all RAW\"\n",
      "\"Do not integrate GSM275288 and GSM275290. Duplicates of this samples already in HS-00661!!!\"\n",
      "\"do not publish\"\n",
      "\"do not publish due to Signal import issues (sample probeset size smaller than expected)\"\n",
      "\"do not publish, strange experimental design\"\n",
      "\"do not release; bad QC – data\"\n",
      "\"do not release; duplicates with HS-00017; HS-00217; HS-00485; HS-01080\"\n",
      "\"do not release; duplicates with HS-00529\"\n",
      "\"do not release; truncated CEL.files –> no QC\"\n",
      "\"double entry with MM-00931\"\n",
      "\"double entry with MM-00963\"\n",
      "\"double staining, Cy3 or Cy5 labeled samples, not supported in GV\"\n",
      "\"Double staining of samples used. GV has not yet supported this type of experiment. Samples GSM846174, GSM846175, GSM846176, GSM846215 do not fit  in ranges <-0.5, 0.5> on RLE. The rest of the samples (23) is OK.\"\n",
      "\"Dual Cy3/Cy5 labeling. Exp. samples - Cy5 label.\"\n",
      "\"dual-labelled samples, it is not supported\"\n",
      "\"Dual-labelled samples, it is not supported in the GV.\"\n",
      "\"Dual-labelled samples, it is not supported in the GV. (The original publication is not available.)\"\n",
      "\"Dual labelling of samples (Cy3, Cy5) is not supported in Genevestigator.\"\n",
      "\"duplicated entry of HS-00456\"\n",
      "\"Duplicated entry with GSE2870 (RN-00004). Data uploaded on GEO and ArrayExpress with the same number of samples, but having a different title.\"\n",
      "\"Duplicated entry with GSE63639 (RN-00251)\"\n",
      "\"duplicated entry with HS-01262\"\n",
      "\"Duplicated entry with HS-01446 (GSE21374)\"\n",
      "\"duplicated entry with HS-02437\"\n",
      "\"duplicated entry with MM-00361\"\n",
      "\"Duplicated entry with MM-00670\"\n",
      "\"Duplicated entry with study MM-00655.\"\n",
      "\"Duplicated entry with superserie GSE9798 (RN-00117)\"\n",
      "\"Duplicate detection revealed 90 duplicate samples, which were previously used in HS-01599 study. The QC plot of other 38 samples is OK.\"\n",
      "\"Duplicated samples were not included\"\n",
      "\"Duplicate entry with MM-00503.\"\n",
      "\"Duplicate entry with MM-00574.\"\n",
      "\"Duplicate Exp.\"\n",
      "\"Duplicate Exp. (annotated as HS-01121)\"\n",
      "\"Duplicate Exp. with HS-00346, HS-00351, HS-00861, HS-01464\"\n",
      "\"duplicate exp. with HS-00602\"\n",
      "\"Duplicate Exp. with HS-01140\"\n",
      "\"Duplicate Exp. with HS-01140 and HS-01485\"\n",
      "\"Duplicate Exp. with HS-01371 as part of superseries\"\n",
      "\"Duplicate Exp. with HS-01449\"\n",
      "\"Duplicate Exp. with HS-01705\"\n",
      "\"Duplicate of Exp. HS-01060: GSM306737 GSM306739 GSM306748 GSM306736 GSM306738 GSM306734 GSM306745 Bad QC: GSM306743 GSM306774 GSM306773 GSM306733\"\n",
      "\"Duplicate of exp HS-01361: GSM542662 GSM542663 GSM542664 GSM542665 GSM542666 Duplicate of exp  HS-01357: GSM542667 GSM542669 GSM542670 GSM542671\"\n",
      "\"Duplicate of HS-01352: GSM647621 GSM647622 GSM647625; Bad QC: GSM647589\"\n",
      "\"duplicate of RN-00198\"\n",
      "\"duplicates detection - the whole experiment (arrays already part of HS-00843, HS-01246, HS-01249)\"\n",
      "\"Duplicates found within the study HS-00970 (GSM19158, GSM19173, GSM20661). HS-00970 is in limbo. Samples were grouped according to platforms. One sample GSM19152 out of 24  from platform  HS_AFFY_U133A was excluded due to low quality data (low correlation, high NUSE).\"\n",
      "\"Duplicates: GSM1334681, GSM1334682, Bad QC: GSM1249377\"\n",
      "\"duplicates: GSM201866,GSM201869,GSM201871,GSM201872,GSM201543,GSM201544,GSM201829,GSM201832,GSM201833,GSM201834; degraded RNA: GSM201838; 13x duplicates of the whole control group\"\n",
      "\"duplicates: GSM395775,GSM395776,GSM395779,GSM395780,GSM395784,GSM395787,GSM395788,GSM395789, GSM395790,GSM523839,GSM523846,GSM523857,GSM523858\"\n",
      "\"Duplicates:GSM549734, GSM549738, GSM549740, GSM549683,GSM549694,, GSM549695, GSM549725, GSM549728, GSM549719, GSM549712, GSM549716, GSM549702, GSM549708, GSM549709,GSM549710\"\n",
      "\"DUPLICATES: GSM606565, GSM606567, GSM606570, GSM606573, GSM606576 (with GSM606565, GSM606567 having bad QC)\"\n",
      "\"Duplicates of Exp. HS-01051: GSM1187246 GSM1187248 GSM1187249 GSM1187251 GSM1187252 GSM1187257 GSM1187259 GSM1187260 GSM1187263 GSM1187265 GSM1187266. Bad QC: GSM1187267\"\n",
      "\"Duplicates to experiment HS-00911 and HS-00506: GSM1683516_HF0512_U133P2, GSM1683520_HF1407_U133P2, GSM1683522_HF1150_U133P2, GSM1683524_HF0963_U133P2, GSM1683527_HF0295_U133P2, GSM1683532_HF0816_U133P2, GSM1683534_HF1356_U133P2, GSM1683537_HF0523_U133P2, GSM1683539_HF0899_U133P2, GSM1683543_HF1589_U133P2, GSM1683544_HF0488_U133P2, GSM1683553_HF0108_U133P2, GSM1683555_HF1490_U133P2, GSM1683557_HF1318_U133P2, GSM1683568_HF1469_U133P2, GSM1683570_HF1235_U133P2, GSM1683585_HF0726_U133P2, GSM1683586_HF0510_U133P2, GSM1683588_HF1540_U133P2, GSM1683592_HF0855_U133P2, GSM1683594_HF1568_U133P2, GSM1683596_HF1382_U133P2, GSM1683597_HF0031_U133P2, GSM1683598_HF1640_U133P2, GSM1683601_HF0244_U133P2, GSM1683603_HF1494_U133P2, GSM1683607_HF1442_U133P2, GSM1683608_HF1000_U133P2, GSM1683610_HF1344_U133P2, GSM1683611_HF0184_U133P2, GSM1683616_HF1702_U133P2, GSM1683619_HF1492_U133P2, GSM1683624_HF1517_U133P2, GSM1683626_HF0434_U133P2, GSM1683629_HF1246_U133P2, GSM1683632_HF1292_U133P2, GSM1683634_HF0954-2_U133P2, GSM1683635_HF0151_U133P2, GSM1683636_HF0087_U133P2, GSM1683637_HF1493_U133P2, GSM1683638_HF0533_U133P2, GSM1683640_HF0300-3_U133P2, GSM1683641_HF1316_U133P2, GSM1683642_HF0936_U133P2, GSM1683645_HF1534_U133P2, GSM1683647_HF0017_U133P2, GSM1683653_HF0520_U133P2, GSM1683657_HF0593_U133P2, GSM1683664_HF1242_U133P2, GSM1683665_HF0992_U133P2, GSM1683673_HF0218_U133P2, GSM1683675_HF1295_U133P2, GSM1683678_E09852_U133P2, GSM1683679_HF1136_U133P2, GSM1683680_HF1581_U133P2, GSM1683687_HF1366_U133P2, GSM1683691_HF0583_U133P2, GSM1683694_HF1338_U133P2, GSM1683696_HF1381_U133P2, GSM1683700_HF1708_U133P2, GSM1683701_HF1191_U133P2, GSM1683706_HF0897_U133P2, GSM1683710_HF0450_U133P2, GSM1683714_HF1269_U133P2, GSM1683722_HF1077_U133P2, GSM1683723_HF0024_U133P2, GSM1683724_HF1409_U133P2, GSM1683726_HF1286_U133P2, GSM1683727_HF0088_U133P2, GSM1683729_HF0850_U133P2, GSM1683730_HF1156_U133P2, GSM1683731_HF1538_U133P2, GSM1683742_HF0543_U133P2, GSM1683748_HF1458_U133P2, GSM1683753_HF0790_U133P2, GSM1683762_HF0285_U133P2, GSM1683763_HF0252_U133P2_2, GGSM1683672_HF0026_U133P2, GSM1683767_HF0138_U133P2, GSM1683772_HF1325_U133P2, GSM1683777_HF1618_U133P2, GSM1683788_HF1167_U133P2, GSM1683793_HF0442-5_U133P2, GSM1683795_HF1502_U133P2, GSM1683798_HF1139_U133P2, GSM1683801_HF0171_U133P2, GSM1683806_HF0639_U133P2, GSM1683820_HF0232_U133P2, GSM1683824_HF0654_U133P2, GSM1683828_HF0089_U133P2, GSM1683833_HF1475_U133P2, GSM1683835_HF1613_U133P2, GSM1683855_HF0526_U133P2, GSM1683857_HF0627_U133P2, GSM1683858_HF0142_U133P2, GSM1683861_HF0982_U133P2, GSM1683864_HF0670_U133P2, GSM1683866_HF0251_U133P2, GSM1683873_HF0616_U133P2, GSM1683874_HF0180_U133P2, GSM1683876_HF0460_U133P2, GSM1683877_HF1097_U133P2, GSM1683880_HF0303_U133P2, GSM1683883_HF0505_U133P2, GSM1683885_HF1489_U133P2, GSM1683889_HF1280_U133P2, GSM1683892_HF1606_U133P2, GSM1683894_HF1058_U133P2, GSM1683896_HF0966_U133P2, GSM1683902_HF0212_U133P2, GSM1683903_HF0894_U133P2, GSM1683908_HF1186_U133P2, GSM1683912_HF0377_U133P2, GSM1683913_HF1628_U133P2, GSM1683914_HF1509_U133P2, GSM1683919_HF1667_U133P2, GSM1683920_HF1232_U133P2, GSM1683921_HF0163_U133P2, GSM1683922_HF0066_U133P2, GSM1683923_HF0953_U133P2_2, GSM1683925_HF1255_U133P2, GSM1683928_HF1587_U133P2, GSM1683930_HF0914_U133P2, GSM1683938_HF0383_U133P2, GSM1683940_HF1227_U133P2, GSM1683943_HF1608_U133P2, GSM1683944_HF1122_U133P2, GSM1683945_HF1397_U133P2, GSM1683946_HF1487_U133P2, GSM1683949_HF0931_U133P2, GSM1683954_HF1511_U133P2, GSM1683961_HF0960_U133P2, GSM1683965_HF0211_U133P2, GSM1683967_HF1677_U133P2, GSM1683973_HF1671_U133P2, GSM1683975_HF0152_U133P2, GSM1683976_HF0316_U133P2, GSM1683983_HF0996_U133P2, GSM1683985_HF0329_U133P2, GSM1684003_HF0291_U133P2, GSM1684004_HF0835_U133P2, GSM1684005_HF0986_U133P2, GSM1684008_HF1178_U133P2, GSM1684016_HF1380_U133P2, GSM1684023_HF1585_U133P2, GSM1684031_HF1334_U133P2, GSM1684032_HF0050_U133P2, GSM1684033_HF1551_U133P2, GSM1684036_HF1262_U133P2, GSM1684041_HF0828_U133P2, GSM1684044_HF1078_U133P2, GSM1684045_HF1137_U133P2, GSM1684046_HF1345_U133P2, GSM1684050_HF0822_U133P2, GSM1684052_HF0962_U133P2, GSM1684053_HF0920_U133P2, GSM1684054_HF0131_U133P2, GSM1684058_HF1057_U133P2, GSM1684061_HF0332_U133P2, GSM1684063_HF0467_U133P2, GSM1684072_HF0975_U133P2, GSM1684073_HF0137_U133P2, GSM1684076_HF0702_U133P2, GSM1684081_HF1357_U133P2, GSM1684088_HF0990_U133P2, GSM1684090_HF0491_U133P2, GSM1684091_HF0120_U133P2, GSM1684092_HF1185_U133P2, GSM1684093_HF1220_U133P2 30 samples not annoateable due to missing information about any anatomy/neoplasm: GSM1683536_03523498_U133P2, GSM1683590_E10476_U133P2, GSM1683605_03525342_U133P2, GSM1683618_E10265_U133P2, GSM1683621_E10595_U133P2, GSM1683631_E10076_U133P2, GSM1683649_E10080_U133P2, GSM1683652_E10494_U133P2, GSM1683688_E10622_U133P2, GSM1683739_03495697_U133P2, GSM1683747_03513138_U133P2, GSM1683768_03514945_U133P2, GSM1683799_03497265_U133P2, GSM1683809_03822538_U133P2, GSM1683810_03822538_U133P2_2, GSM1683826_E10287_U133P2, GSM1683846_03089254_U133P2, GSM1683847_E10692_U133P2, GSM1683865_03513467_U133P2, GSM1683955_E09960_U133P2, GSM1683957_81668782_U133P2, GSM1683972_03506280_U133P2, GSM1683988_03522254_U133P2, GSM1684002_03504050_U133P2, GSM1684011_E10136_U133P2, GSM1684029_E10210_U133P2, GSM1684042_00860772_U133P2, GSM1684055_E10235_U133P2, GSM1684089_E10253_U133P2, GSM1684094_03512529_U133P2 Bad QC: GSM1683605_03525342_U133P2, GSM1683717_Normal_1_U133_P2, GSM1683718_E09893_U133P2, GSM1683719_E09848_U133P2, GSM1683739_03495697_U133P2, GSM1683747_03513138_U133P2, GSM1683749_E10514_U133P2, GSM1683750_E10216_U133P2, GSM1683796_E09513_U133P2, GSM1683809_03822538_U133P2, GSM1683810_03822538_U133P2_2, GSM1683816_E10252C_NT_U133P2, GSM1683850_E09846_U133P2, GSM1683867_E10103_U133P2, GSM1683878_HF0757_U133P2, GSM1683907_E09430_U133P2, GSM1683988_03522254_U133P2, GSM1683995_E10488_U133P2, GSM1683998_HF0178_U133P2, GSM1684021_HF1032_U133P2, GSM1684043_621233_U133P2\"\n",
      "\"DUPLICATES with CL-00023: GSM559095, GSM559097, GSM559099, GSM559101, GSM559103, GSM559081, GSM559083, GSM559085, GSM559087, GSM559089, GSM559091\"\n",
      "\"Duplicates with exp. HS-00247: GSM244488,GSM244489,GSM244490,GSM244491,GSM244492,GSM244493,GSM244494,GSM244495,GSM244496,GSM244497,GSM244498,GSM244507,GSM244508,GSM244509,GSM244510 --> All Con samples...therefore exp. was merged. Bad QC: GSM244499\"\n",
      "\"duplicates with HS-00667\"\n",
      "\"duplicates (with HS-00838): GSM1025129_E647JJ, GSM1025130_E648JJ, GSM1025131_E650JJ, GSM1025132_E652JJ, GSM1025133_E654JJ, GSM1025134_E655JJ, GSM1025135_F745JJ, GSM1025136_F747JJ, GSM1025137_F749JJ, GSM1025138_F750JJ, GSM1025139_F751JJ, GSM1025140_F803JJ, GSM1025141_F804JJ, GSM1025152_E241JJ, GSM1025172_F788JJ, GSM1025176_F792JJ; excluded based on QC: GSM1025195_F781JJ\"\n",
      "\"duplicates with HS-01295 not included\"\n",
      "\"Duplicates (with HS-01360): GSM377062, GSM377063, GSM377064, GSM377066, GSM377067, GSM377068, GSM377069, GSM377065; check of anatomy annotation on 41-vrt revealed that the other two groups from this experiment have different anatomy (PBMC x whole blood); conclusion - do not publish the experiment\"\n",
      "\"Duplicates with HS-01430: GSM709349, GSM709350, GSM709351, GSM709352, GSM709353, GSM709354\"\n",
      "\"duplicates (with HS-01455): GSM1310919, GSM1310920, GSM1310921, GSM1310922\"\n",
      "\"Duplicates with MM-00281\"\n",
      "\"duplicates with MM-00324\"\n",
      "\"Duplicates with samples in HS-01373\"\n",
      "\"duplication of HS-00412\"\n",
      "\"Duplication. Samples are analyzed under GSE1133.\"\n",
      "\"Duplictaes with HS-00602 and HS-01555\"\n",
      "\"E-MTAB-3012 contains dual-labelled samples.It is not supported in GV.\"\n",
      "\"Entire experiment is part of HS-00733. Thus, not to be integrated!\"\n",
      "\"E-TABM-208 contains dual-labelled samples.It is not supported in GV.\"\n",
      "\"E-TOXM-17-raw-cel-591949540 excluded, all other samples have a good quality.\"\n",
      "\"excluded based on QC: GSM181407\"\n",
      "\"excluded based on QC: GSM205803, all other samples have a good quality.\"\n",
      "\"excluded based on QC: GSM226758\"\n",
      "\"excluded based on QC: GSM393550\"\n",
      "\"excluded based on QC: GSM491307, excluded based on duplicate: GSM491319\"\n",
      "\"excluded based on QC: GSM518243\"\n",
      "\"excluded based on QC: GSM650510, GSM650521, GSM650536, GSM650621, GSM650632, GSM650653; excluded from control group due to co-morbidity not stated in an article: GSM650646, GSM650600\"\n",
      "\"excluded based on QC: GSM650725, GSM650726, GSM650766, GSM650824, GSM650827, GSM650830, GSM650834, GSM650854; excluded due to birth defects: GSM650825, GSM650709, GSM650692, GSM650722\"\n",
      "\"excluded based on QC: P15015_920408_S-000031254_20140723.CEL, P15015_921046_S-000033354_20140723.CEL, P15015_920908_S-000033388_20140611.CEL\"\n",
      "\"excluded base on QC: GSM661732, GSM661734, GSM661737, GSM661768, GSM661788\"\n",
      "\"excluded due to inconsistencies and unclear anatomy (HGPIN or adenocarcinoma) and if tissue really neoplastic ....\"\n",
      "\"excluded due to low quality ( RNA degradation, array-array correlation ~0.2, high RLE and NUSE): GSM2033074_WT_H2O_6609, GSM2033075_WT_H2O_6610, GSM2033076_WT_H2O_6616,  GSM2033077_WT_DSS_6617,  GSM2033078_WT_DSS_6618\"\n",
      "\"excluded due to unclear annotation: GSM748877, GSM748889, GSM748876, GSM748888, GSM748878, GSM748890\"\n",
      "\"excluded due to unclear annotation in repository: GSM948577, GSM948578\"\n",
      "\"excluded duplicates (with HS-01362): GSM928885, GSM928886, GSM928887, GSM928888, GSM928889, GSM928890, GSM928891, GSM928892; excluded based on QC (bad array-array correlation): GSM928845, GSM928854, GSM928864, GSM928830\"\n",
      "\"Excluded samples from annotation: GSM606126, GSM606130 because of low QC. The rest of samples with good QC.\"\n",
      "\"Excluded samples: GSM1174050; GSM1174053; GSM1174061; GSM1174122; GSM1174154; GSM1174094; GSM1174116; GSM1174123 because of low quality.\"\n",
      "\"Excluded samples: GSM1381437, GSM1381438 and GSM1381531 because of border RLE and NUSE with correlation around 0.72 - 0.8.\"\n",
      "\"excluded samples: GSM2226282 - due to high NUSE values and low correlation with other samples. Three other samples (GSM2226338, GSM2226357 and GSM2226401) have elevated NUSE values and lower correlations, but acceptable.\"\n",
      "\"excluded samples GSM252847, GSM252843, GSM252845 because of duplication in HS-01926\"\n",
      "\"Exclusion of sample GSM2399840 due to low correlation (0.74). The other samples have a good quality. Cy3/Cy5 labeling used, only Cy3 labeled samples curated. GV does not support Cy5 labeling.\"\n",
      "\"Experiment annotated but cel. files are truncated\"\n",
      "\"experiment containes two platforms, both corresponding to U1333A but with different CDFs; each sample is present on both platforms; RAW signals were detected as duplicates and were therefore excluded (31 samples)\"\n",
      "\"Experiment has been rejected due to low quality of the samples (expression intensity below 6; low Array-Array correlation: bellow 0.65)\"\n",
      "\"experiments with two platfroms, only GPL96 samples anniotated (17 samples, 1x excluded based on QC: GSM251542), GPL571 (36 samples) not annotated\"\n",
      "\"Experiment was deleted from the database due to low quality of signal data.\"\n",
      "\"Experiment was removed from public database due to low quality of signal data. Low - medium-high expressed genes had expression more less in the same range, and did not correspond to the behavior of the ref. genes in other experiments.\"\n",
      "\"Experiment was removed from the database due to very low quality of signal data.\"\n",
      "\"experiment was unpublished - based on gender markers, there is a strong evidence that the cell line was contaminated, it is a male cell line, but some samples have female signature with gender markers (XIST, RPS4Y1)\"\n",
      "\"Experiment with 2 platform: GPL97 - platform is not supported; samples on GPL96 (115 samples) were detected as duplicates : 112 samples of HS-01464 have duplicates with HS-00861 (112 )and also with HS-00346 (42). 3 samples are exclusively in HS-01464. That is the reason for rejecting the experiment\"\n",
      "\"experiment without signal data\"\n",
      "\"experiment with two platforms, only samples from GPL6480 were curated\"\n",
      "\"F1F2WTFAST−1, F1F2WTFAST−2,  F1F2WTFED−1, F1F2WTFED−2, KO1KO2WTFAST−1, KO1KO2WTFAST−2, KO1KO2WTFED−1, KO1KO2WTFED−2 excluded due to high nuse, low correlation\"\n",
      "\"GSE11805 number was added to the already existing MM-00376.\"\n",
      "\"GSE18612 contains dual-labelled samples.It is not supported in GV.\"\n",
      "\"\"GSE31983\" was deleted by the GEO staff from GEO webpage. Samples from the HS-00908 were duplicate detected in the study HS-01476 (GSE47969). Moreover new sample characteristics are available for GSE47969. Therefore new annotation of HS-01476 was performed.\"\n",
      "\"GSE58336 contains dual-labelled samples.It is not supported in GV.\"\n",
      "\"GSM1004101 GSM1004112\"\n",
      "\"GSM1024347, GSM1024358, GSM1024359, GSM1024364 excluded due to high NUSE. All other samples have a good quality.\"\n",
      "\"GSM1063383 bad QC: deleted\"\n",
      "\"GSM1065751_lung_27; deleted bad QC (degradation)\"\n",
      "\"GSM1108153, GSM1108158, GSM1108169 bad QC\"\n",
      "\"GSM1146320_H7-Glom-TN995.CEL.gz, GSM1146321_H7-Glom-TN996.CEL.gz, GSM1146322_H7-Glom-TN997.CEL.gz, GSM1146427_H7-Tub-TN1185.CEL.gz, GSM1146428_H7-Tub-TN1186.CEL.gz, GSM1146429_H7-Tub-TN1187.CEL.gz, GSM1146312_H7-Glom-RPGN935.CEL.gz; samples from the tumor nephrectomy were not included, as it was not clear if they took healthy tissue or tumor tissue\"\n",
      "\"GSM1146323_H1-Tub-CD1.CEL.gz, GSM1146324_H1-Tub-CD2.CEL.gz, GSM1146325_H1-Tub-CD3.CEL.gz, GSM1146326_H1-Tub-CD4.CEL.gz, GSM1146366_H5-Tub-HT11.CEL.gz, GSM1146367_H5-Tub-IgA27.CEL.gz, GSM1146369_H5-Tub-MGN303.CEL.gz, GSM1146368_H5-Tub-MCD8.CEL.gz, GSM1146215_H1-Glom-TN2.CEL.gz, GSM1146216_H1-Glom-TN4.CEL.gz, GSM1146217_H1-Glom-TN996.CEL.gz, GSM1146218_H1-Glom-TN997.CEL.gz, GSM1146219_H2-Glom-TN2.CEL.gz, GSM1146220_H2-Glom-TN4.CEL.gz, GSM1146221_H2-Glom-TN996.CEL.gz, GSM1146222_H2-Glom-TN997.CEL.gz, GSM1146229_H3-Glom-TN2.CEL.gz, GSM1146230_H3-Glom-TN4.CEL.gz, GSM1146231_H3-Glom-TN995.CEL.gz, GSM1146232_H3-Glom-TN996.CEL.gz, GSM1146233_H3-Glom-TN997.CEL.gz, GSM1146234_H3-Glom-TN997_1.CEL.gz, GSM1146353_H3-Tub-TN1185.CEL.gz, GSM1146354_H3-Tub-TN1186.CEL.gz, GSM1146355_H3-Tub-TN2.CEL.gz; samples from the tumor nephrectomy were not included, as it was not clear if they took healthy tissue or tumor tissue\"\n",
      "\"GSM1150070_LT160423LL_CTRL, GSM1150041_LT122757RU_CTRL, GSM1150147_LT030041RU_COPD, GSM1150499_LT233821RL_COPD, GSM1150547_LT299181LU_ILD; bad QC deleted\"\n",
      "\"GSM1151109: bad QC\"\n",
      "\"GSM1152882; deleted due to bad QC\"\n",
      "\"GSM1192714_BR56_4SN.CEL.gz excluded, low array-array correlation, high NUSE (1.1)\"\n",
      "\"GSM1202434, \\\n",
      "\"GSM1218142: Bad correlation, other samples have a good quality.\"\n",
      "\"GSM1224963, GSM1224966: bad QC\"\n",
      "\"GSM1241878_29nad122 and GSM1241878_29nad119 excluded due to high NUSE and bad clustering.\"\n",
      "\"GSM1289688_FLU00044.day0; deleted due to bad QC\"\n",
      "\"GSM1401308, GSM1401313, GSM1401323\"\n",
      "\"GSM1466225_siFoxo3_03 excluded, all other samples have a good quality.\"\n",
      "\"GSM1490092_wk11_wt_6.CEL.gz, GSM1490095_wk15_SOD_3.CEL.gz excluded due to high NUSE and low correlation.\"\n",
      "\"GSM1529854_M32a, GSM1529855_M37 were excluded based on high NUSE. Other samples have good quality.\"\n",
      "\"GSM15849 excluded, all other samples have a good quality.\"\n",
      "\"GSM1592676, GSM1592677, GSM1592691, GSM1592695 excluded (based on RLE, NUSE, and Array-Array correlation)\"\n",
      "\"GSM1616397, GSM1616399, GSM1616401, GSM1616402, GSM1616404, GSM1616407, GSM1616408, GSM1616410 not uploaded to GEO, Other samples have a good quality.\"\n",
      "\"GSM1644139_Woodling_wt_ibuprofen_1_MoGene−1_0−st−v1 and GSM1644143_Woodling_wt_ibuprofen_5_MoGene−1_0−st−v1 excluded due to high NUSE.\"\n",
      "\"GSM16507 excluded, all other samples have a good quality.\"\n",
      "\"GSM1673676-GSM1673678 excluded due to low quality.\"\n",
      "\"GSM1680804_TAK1_BCR_Batch1 and GSM1680858_DMSO_BCR_Batch1 excluded due to high NUSE, low correralation.\"\n",
      "\"GSM181381.CEL.gz  and GSM181382.CEL.gz are identical\"\n",
      "\"GSM185021.CEL: Bad NUSE (>1.10), correlation in comparison to others \"low\" (still >0.8). Kept. Check in GV after import.\"\n",
      "\"GSM189787 bad QC\"\n",
      "\"GSM200581 excluded due to high NUSE and low correlation.\\\n",
      "\"GSM201487, GSM201488, GSM201489 excluded by Oliver\"\n",
      "\"GSM21045, GSM21046 excluded, all other samples have a good quality.\"\n",
      "\"GSM231435 excluded - CEL file truncated\"\n",
      "\"GSM243353\\\n",
      "\"GSM27002 excluded, all other samples have a good quality.\"\n",
      "\"GSM27446, GSM27447 excluded. All other samples have a good quality.\"\n",
      "\"GSM34331 excluded. All other samples have a good quality.\"\n",
      "\"GSM346581, GSM346572\"\n",
      "\"GSM351073, GSM351074, GSM351075 excluded notenogh info to annotate.\"\n",
      "\"GSM358110 excluded due to poor quality. GSE14323 study contains two different platforms, therefore it was split into two different HS-numbers.\"\n",
      "\"GSM4018 excluded, all other samples have a good quality.\"\n",
      "\"GSM4119 excluded, all other samples have  a good quality.\"\n",
      "\"GSM442553, GSM442560 were excluded due to high NUSE. All other samples are O.K.\"\n",
      "\"GSM44503 excluded, all other samples have a good quality.\\\n",
      "\"GSM458011, GSM457996, GSM458001 - bad QC (low correlation)\"\n",
      "\"GSM468333, GSM468341, GSM468352, GSM468375\"\n",
      "\"GSM505332 and GSM505302: not clear to which group they belong to\"\n",
      "\"GSM533867 bad correlation\"\n",
      "\"GSM533888_041508JP-4, GSM533894_041508JP-12, GSM533897_041508JP-16; deleted due to bad QC\"\n",
      "\"GSM575927; deleted due to bad QC\"\n",
      "\"GSM595745, GSM595758, GSM595771, GSM595734, GSM595801, GSM595776, GSM595738, GSM595725, GSM595714, GSM595783\"\n",
      "\"GSM620772, GSM620773, GSM620792, GSM620788, GSM620793; deleted due to bad QC\"\n",
      "\"GSM624937, GSM624971, GSM624938, GSM624974, GSM624970\"\n",
      "\"GSM63285 sample excluded, all other samples have a good quality.\"\n",
      "\"GSM691122 - bad QC; GSM691134 - duplication with HS-00853 (GSM465303)\"\n",
      "\"GSM693824; deleted due to bad format: (after deletion QCNM worked)\"\n",
      "\"GSM738331_FJS319_Goldman_U133Plus_2 excluded due to high nuse and low correlation. All other samples have good quality\"\n",
      "\"GSM740106 border quality, but not excluded, Other samples have a good quality.\"\n",
      "\"GSM756539_TOV21G_control excluded, All other samples have a good quality.\"\n",
      "\"GSM757027_KS1-HG_U133A_2-2903 bad QC: deleted; 25 control samples excluded due to intra-experiment duplication. This duplication is labeled as \"reanalysis\" in GEO matrix file.\"\n",
      "\"GSM815345_SAMPLE_150.txt.gz - removed wrong sample format; GSM815233_SAMPLE_38.txt.gz,GSM815214_SAMPLE_19.txt.gz,GSM815287_SAMPLE_92.txt.gz,GSM815252_SAMPLE_57.txt.gz,GSM815265_SAMPLE_70.txt.gz\"\n",
      "\"GSM830077_PBB_251485059414_S01-2_GE1_107_Sep09_1_2, GSM829996_PBB_251485060446_S01-2_GE1_107_Sep09_1_4, GSM830021_PBB_251485059132_S01_GE1_107_Sep09_1_1, GSM830000_PBB_251485060447_S01-2_GE1_107_Sep09_1_1; deleted due to bad QC\"\n",
      "\"GSM839243_Pieter_45 bad QC: deleted\"\n",
      "\"GSM866325 excluded based on QC, NUSE, A-A\"\n",
      "\"GSM873834_HC1369_pbmc727_9hr_40_091609: high positive border elements but otherwise ok; not deleted - checked in GV; GSM1274686_MK6_34_RO+IL1ra_2751_072210.CEL.gz /outlayer in QC; GSM1274682_21_ro+IL1ra_2393_727_7-14-10- no infromation about subject\"\n",
      "\"GSM899056, GSM899127, GSM899155, GSM899079, GSM899254 bad QC\"\n",
      "\"GSM92645 excluded. All other samples have a good quality.\"\n",
      "\"GSM927649_S11-CEL37\"\n",
      "\"GSM944816_Let−7−1.txt.gz, GSM944828_let−7−mir30b−3.txt; deleted due to bad QC\"\n",
      "\"GSM981323 and GSM981324 bad QC\"\n",
      "\"GSM98566 excluded, all other samples have a good quality.\"\n",
      "\"GV unsupported Cy5/Cy3 staining.\"\n",
      "\"has less probesets than expected\"\n",
      "\"Hyb1.CEL\"\n",
      "\"Illumina. Cannot be integrated in GV, deliver as a flat file. QC results: very low array-array correlations, no sample meets the 0.8 threshold. Do not have any other QC reports for experiments on this platform, so cannot judge the QC. Wait for more QC results or deliver as a flat file without a QC assessment.\"\n",
      "\"Internally duplicated entry\"\n",
      "\"Internally duplicated entry → REJECTED\"\n",
      "\"in vitro study in HEK cells, from the annotations is not clear, if it was knock down or overexpression of target protein.\"\n",
      "\"issue with signal data processing ( no TXT - files)\"\n",
      "\"low array-array correlation: GSM1179967_Pre-IFX-7, GSM1179978_Pre-IVIG-2, GSM1179980_Pre-IVIG-4\"\n",
      "\"low array-array correlation: GSM1411029_Turner_14_B_control_HG−U133_Plus_2_\"\n",
      "\"low array-array correlation: GSM214221; GSM214223\"\n",
      "\"low array-array correlation: GSM347816; GSM347817; GSM347828\"\n",
      "\"low array-array correlation: GSM421927, GSM421928; duplicates with HS-01331: GSM421901.CEL.gz, GSM421904.CEL.gz, GSM421903.CEL.gz, GSM421902.CEL.gz\"\n",
      "\"low array-array correlation: GSM591061\"\n",
      "\"low array-array correlation: GSM663743\"\n",
      "\"Low correlation among the samples, 0.45.\"\n",
      "\"Low correlation in some samples (0.65).\"\n",
      "\"Low quality data, correlation 0.1.\"\n",
      "\"Low quality of all samples\"\n",
      "\"Low quality of most of the samples\"\n",
      "\"Low quality of overall signal data\"\n",
      "\"low quality of signal data\"\n",
      "\"Low quality of signal data (2 peaks on \"expression density plot\").\"\n",
      "\"Many samples have strange QC (higher expression boxplot (6-9), low positive border elements (1000-2000)) but other parameters are ok. Nevertheless all samples were annotated and were checked and was decided to keep them all.\"\n",
      "\"merged to HS-00838\"\n",
      "\"merged to RM-00005\"\n",
      "\"merged to RM-00013\"\n",
      "\"merged with HS-01711 (partial duplicates)\"\n",
      "\"Most of the samples are OK. Overall low quality samples-  GSM1194576, GSM1194577, GSM1194580, GSM1194587, were excluded from further analysis.\"\n",
      "\"multiple platform and multiple organism experiment; human (18 samples): all duplicates of samples included in HS-00924; Pan troglodytes: hybridized on GPL570 human chip, therefore not annotated; monkey data - txt file produced but QC report was not written\"\n",
      "\"multiple platform exp.\\\n",
      "\"Multiple Platform Experiment: GPL1074, GPL1073, GPL96\"\n",
      "\"Multiple Platform Experiment: GPL1261, GPL570\"\n",
      "\"Multiple Platform Experiment: GPL1426, GPL570\"\n",
      "\"Multiple Platform Experiment: GPL2507, GPL96\"\n",
      "\"Multiple Platform Experiment: GPL3282, GPL96\"\n",
      "\"Multiple Platform Experiment: GPL339, GPL96\"\n",
      "\"Multiple Platform Experiment: GPL570, GPL97, GPL96\"\n",
      "\"Multiple Platform Experiment: GPL91, GPL81\"\n",
      "\"Multiple Platform Experiment: GPL93, GPL92, GPL91\"\n",
      "\"Multiple Platform Experiment: GPL96, GPL91\"\n",
      "\"Multiple Platform Experiment: GPL96, GPL97\"\n",
      "\"Multiple Platform Experiment: GPL96, GPL97,GPL570\"\n",
      "\"Multiple Platform Experiment: GPL97, GPL96\"\n",
      "\"multiple platform experiment, only data from GPL1261 were curated\\\n",
      "\"multiple platform experiment, only data from GPL81 were curated\\\n",
      "\"multiple platform exp., only data from GPL1261 were curated\\\n",
      "\"multiple platform exp.; only data from GPL1261 were curated\\\n",
      "\"Multiple platform exp.; only data from GPL1261 were curated\"\n",
      "\"multiple platform exp.; only data from GPL81 were curated\"\n",
      "\"multiple platform exp.; only data from GPL81 were curated\\\n",
      "\"multiple platform exp.; only data from GPL81 were curated, all samples have  a good quality.\"\n",
      "\"multiple platform exp.; RAW data not available\"\n",
      "\\N\n",
      "\"NCI-H460_Spheres_L removed due to negative values\"\n",
      "\"NM/QC failed, import into GENEVESTIGATOR is not possible.\"\n",
      "\"No Bad QC, but MDA-MB-435 cell line with corresponding CEL not in annotation (Cell line iGSM803744_118180hp133a11s reanalysed to melaonma cell line): GSM803626_113395hp133a11, GSM803685_113454hp133a11,\"\n",
      "\"NO correct RAW\"\n",
      "\"No data available for download.\"\n",
      "\"Non-normalized data not available. Study is available as a flat file.\"\n",
      "\"No non-normalized data. Available as flat file. Study was requested by PMI, but curation was cancelled due to higher PMI priorities.\"\n",
      "\"No raw data.\"\n",
      "\"No RAW data\"\n",
      "\"no raw data available\"\n",
      "\"no RAW data available\"\n",
      "\"No raw data available.\"\n",
      "\"No raw data available. Downloaded file could not be processed.\"\n",
      "\"No  RAW for GSM912836-GSM912875 (40 healthy samples); Bad NUSE and Correlation:GSM912790_H641\"\n",
      "\"No signal data.\"\n",
      "\"No Signal - Data: GSM1219468_Mkl-1_1_CSAS12299, GSM1219469_Mkl-1_2_CSAS12299, GSM1219470_Mkl-1_3_CSAS12299, GSM1219471_Mkl-1_1_CSAS13317, GSM1219472_Mkl-1_2_CSAS13317, GSM1219473_Mkl-1_3_CSAS13317, GSM1219474_UISO_1_CSAS12299, GSM1219475_UISO_2_CSAS12299, GSM1219476_UISO_3_CSAS12299, GSM1219477_UISO_late_1, GSM1219478_UISO_late_2, GSM1219479_UISO_late_3, GSM1219480_WaGa_1_CSAS12299, GSM1219481_WaGa_2_CSAS12299, GSM1219482_WaGa_3_CSAS12299, GSM1219483_WaGa_1_CSAS13317, GSM1219484_WaGa_2_CSAS13317, GSM1219485_WaGa_3_CSAS13317, GSM1219486_MC01_HG-U133A_2, GSM1493332_MCC26_p11_CSAS16385, GSM1493333_MCC26_p12_CSAS16385, GSM1493334_MCC26_p13_CSAS16385, GSM1493335_MCC13_p36_CSAS16385, GSM1493336_MCC13_p37_CSAS16385, GSM1493337_MCC13_p38_CSAS16385, GSM1493338_Waga_CSAS16385, GSM1493339_U_late_CSAS16385, GSM1219442_MT07_HG-U133A_2, GSM1219466_LT08_HG-U133A_2, GSM1219467_LT11_HG-U133A_2\"\n",
      "\"Not processed due to Illumina issue with random missing probesets.\"\n",
      "\"not public, RMA Data missing\"\n",
      "\"no txt, no QC generated\"\n",
      "\"One CEL file, GSM619243, was truncated and could thus not be included\"\n",
      "\"One sample, GSM1525339_EA10023_104588_Medicago_10DPPD_10DD_, was excluded as it might have been mislabeled.\"\n",
      "\"Only 17 samples on supported platform, could be annotated on customer request.\"\n",
      "\"Only annotation. Low quality data,  low signal, small differences between positive and negative border elements, unusual RNA degradation plots.\"\n",
      "\"Only C3 labelled samples were included into the annotation. Samples: WIS_251486835147_S01_GE2-v5_95_Feb07_1_1.txt, WIS_251486835147_S01_GE2-v5_95_Feb07_1_4.txt, WIS_251486835148_S01_GE2-v5_95_Feb07_1_1.txt, WIS_251486835149_S01_GE2-v5_95_Feb07_1_4.txt, were exclude due to overall low quality.\"\n",
      "\"Only pancreas T cells samples included.\"\n",
      "\"Overall lower correlation 0.7 among samples.\"\n",
      "\"Overall low quality data.\"\n",
      "\"overall low signal intensity, otherwise all samples OK\"\n",
      "\"Part of the data are in unusual format. Only samples with correct data format were analyzed.\"\n",
      "\"Part of the samples are on Cy5.\"\n",
      "\"plattform can still not be used (+QC probl)\"\n",
      "\"Problems with date processing, txt files are missing, QC did not perform.\"\n",
      "\"Problem with downloading of raw-data.\"\n",
      "\"Problem with downloading of raw data for sample GSM2038228. Moreover sample GSM2038233 was excluded due to bad QC (low RLE, low array-array correlation).\"\n",
      "\"Problem with downloading of raw data. Therefore only 26 samples were analyzed.\"\n",
      "\"QC: All samples OK. Metformin treated samples (GSM1127804 - GSM1127814; GSM1127826 - GSM1127836; GSM1127848 - GSM1127858) excluded from the study due to missing information regarding metformin treatment (dose, time). These samples were omitted in original publication, as well.\"\n",
      "\"QC failed.\\\n",
      "\"QC failed\"\n",
      "\"QC failed. Study is available as a flat file.\"\n",
      "\"QC failed. Study is available as flat file.\"\n",
      "\"QC failed. The study is available as a flat file.\"\n",
      "\"QC fails - negative values in signal. \\\n",
      "\"QC for all samples is OK, but  almost all are duplicates (subserie), except of sample GSM873748.\"\n",
      "\"QC is OK, but samples are duplicates from study HS-01599.\"\n",
      "\"QC is OK, but samples are duplicates (subserie) from HS-01045 study.\"\n",
      "\"QC issues repeatedly; not curated\"\n",
      "\"QCNM failed\"\n",
      "\"QCNM failed, customized chip which is not equal to AGIL_4x44K as they claimed\"\n",
      "\"QCNM failed (Illumina pipeline issue)\"\n",
      "\"QCNM failed (Illumina platform)\"\n",
      "\"QCNM failed, RAW files corrupted\"\n",
      "\"QCNM failed, RAW files corrupted\\\n",
      "\"QCNM failed; third-party re-analysis experiment\"\n",
      "\"QCNM not available. Curation available as flat file.\"\n",
      "\"QCNM not available (currently unsupported platform).\"\n",
      "\"QCNM not available (currently unsupported platform)\"\n",
      "\"QCNM not available (unsupported platform). Curation available as flat file.\"\n",
      "\"QCNM ready (duplicates with HS-00412 and HS-01402)\"\n",
      "\"QCNM ready (duplicates with HS-01401)\"\n",
      "\"QC not available, not working platform, add QC later\"\n",
      "\"QC of all samples is OK. Sample GSM762814 has not clear description (name).\"\n",
      "\"QC of all samples ok.\"\n",
      "\"QC of all samples OK. Poor description of samples in both Publication and Matrix file. In the Publication - anatomy: renal cortex, in the matrix file: kidney / primary cell.\"\n",
      "\"QC of samples is OK.\"\n",
      "\"QC ok.\"\n",
      "\"QC O.K.\"\n",
      "\"QC ok for samples on GPL570, GPL96 and 97 with one sample each (GSM1060794 and GPL106ß795); both not annotated Wrong Signaldata - format: GSM1060754_Adipose_U133p2\"\n",
      "\"QC plots are good.\"\n",
      "\"QC problem\"\n",
      "\"QC report failed 2x, but txt files are produced, DC on 41-vrt - all OK\"\n",
      "\"QC results not very good, but acceptable.\"\n",
      "\"QC samples are ok.\"\n",
      "\"QC samples are OK.\"\n",
      "\"QC samples are OK; 23 from 24 samples are duplicates (subserie) - to HS-01178 study (GSE42488).\"\n",
      "\"QC samples are ok. All samples are duplicates (subserie) to HS-01178 study (GSE42488).\"\n",
      "\"QC samples are relatively ok. Sample GSM764147 has low quality in NUSE graph.\"\n",
      "\"Quite bad quality of all samples\"\n",
      "\"Raw data are not available.\"\n",
      "\"RAW data available in the GEO repository were not generated on Agilent platform and were pre-processed before loading to GEO. The data can't be processed in the internal pipeline.\"\n",
      "\"RAW data corrupted. Experiment is fully annotated, but can't be imported into GENEVESTIGATOR.\"\n",
      "\"raw data files not provided\"\n",
      "\"Raw data format not compatible with our pipeline.\"\n",
      "\"raw data not available\"\n",
      "\"Raw data not available.\"\n",
      "\"Raw data not available\"\n",
      "\"RAW data not available.\"\n",
      "\"RAW data not available\"\n",
      "\"RAW -files corupt\"\n",
      "\"Rejected because 208 samples are duplicates with samples which have been already imported into the GV.\"\n",
      "\"rejected because all samples are part / duplicates of existing experiment HS-00598 (GSE16879)\"\n",
      "\"rejected due to duplicated samples with HS-00149 and HS-00184\"\n",
      "\"Rejected due to lack of information - stable/non-stable transfection, missing publication, unknown study design, not possible to state cell line/primary cell\"\n",
      "\"rejected due to low quality of signal data\"\n",
      "\"rejected due to overall low quality of signal data\"\n",
      "\"Rejected due to re-analyses from GSE12948.\"\n",
      "\"Removed samples due to bad QC:\\\n",
      "\"Replicate Exp (RM-00012)\"\n",
      "\"Replicate exp (RN-00200)\"\n",
      "\"Replicate of Exp. HS-01291: GSM912199, GSM912203, GSM912206, GSM912209, GSM912236, GSM912255, GSM912273, GSM912282, GSM912291, GSM912304, GSM912309, GSM912310, GSM912313, GSM912320, GSM912330, GSM912342, GSM912355, GSM912363, GSM912364, GSM912403, GSM912411, GSM912413, Bad QC: GSM912300, GSM912463, GSM912486\"\n",
      "\"Replicates of exp. HS-00519, HS-00247, HS-00391: GSM648591 GSM648592 GSM648593 GSM648594 GSM648595 GSM648596 GSM648597 GSM648598 GSM648599GSM648600 GSM648601 GSM648602 GSM648603 GSM648615 GSM648616 GSM648617 GSM648618 GSM648619 GSM648620 GSM648621 GSM648622 GSM648623 GSM648624 GSM648625 GSM648626 GSM648633 GSM648634 GSM648635 GSM648636 GSM648637 GSM648639 GSM648642GSM648643 GSM648644 GSM648645 GSM648647 GSM648648 GSM648649 GSM648650 GSM648651 GSM648652 GSM648653 GSM648655 GSM648656 GSM648657 GSM648658 GSM648659 GSM648660 GSM648666 GSM648667 GSM648668 GSM648669 GSM648670 GSM648671 GSM648672 GSM648673 GSM648674 GSM648675 GSM648676 GSM648677 GSM648678 GSM648679 GSM648680 GSM648681 GSM648682 GSM648687 GSM648695 GSM648696 GSM648697 GSM648698 GSM648699 GSM648700 GSM648701 GSM648702 GSM648703 GSM648704 GSM648705 GSM648707 GSM648708 GSM648709 GSM648715 GSM648718 GSM648719, and GSM647621 GSM647622 GSM647625 Bad QC: GSM648706 GSM648590 GSM647589\"\n",
      "\"Samles are OK. Sample GSM1525583 has been excluded due to lower quality.\"\n",
      "\"Sample 71 and 72 bad QC\"\n",
      "\"Sample A120Het1 excluded (low correlation 0.72, NUSE above 1.1). The other samples are OK\"\n",
      "\"Sample George1912−1 was excluded due to low quality ( NUSE above 1.1). Other samples are OK.\"\n",
      "\"Sample GSM1064755 was excluded because of low QC. The rest of samples is with good QC. The study publication is not available.\"\n",
      "\"Sample GSM1070442 was excluded due to lower array-array correlation (QC) and low correlation matrix value in GV.\"\n",
      "\"Sample GSM108567 was excluded, the rest of samples were good quality.\"\n",
      "\"Sample GSM1129972 excluded for low quality.\"\n",
      "\"Sample GSM1130670 excluded (strange format of raw data). Other samples have a good quality.\"\n",
      "\"Sample GSM115697 was excluded, the rest of samples were good quality.\"\n",
      "\"Sample GSM1163791_DUKE38_334.CEL.gz was excluded due to high NUSE and lower array-array correlation.\"\n",
      "\"sample GSM1224911_rota_seizure_2_acute.CEL.gz has worse correlation with others but still acceptable\"\n",
      "\"Sample GSM1250699 is excluded due to duplicate detection. Sample GSM1250706 is excluded due to high NUSE and lower array-array correlation.\"\n",
      "\"Sample GSM147516 was excluded from further analysis. The rest of samples are good quality.\"\n",
      "\"Sample GSM1490136 excluded for low quality data. Other samples are of various quality, but still acceptable.\"\n",
      "\"Sample GSM1534034 was excluded due to the low quality (NUSE plot higher centered/large spread box, low correlation 0.74). Other samples are OK.\"\n",
      "\"sample GSM1544490_DNA9183−017_HG−U133_Plus_2_ is excluded, RNA degradation and low aray-aray correlation\"\n",
      "\"Sample GSM158619 excluded, all other samples have a good quality.\"\n",
      "\"Sample GSM1627762_NAS_72 was excluded due to low array-array correlation.\"\n",
      "\"Sample GSM1700965 is excluded from the study annotation, because of low QC quality.\"\n",
      "\"Sample GSM1849811 excluded for low quality data.\"\n",
      "\"Sample GSM1902271 has low value in correlation plot. The rest of samples are ok.\"\n",
      "\"Sample GSM1916453 excluded for low quality data.\"\n",
      "\"Sample GSM2026127 excluded due to low quality (expression density plot-strange shape, correlation 0.65, RLE above 0.5), other samples are OK.\"\n",
      "\"Sample GSM2101434_S9_HG−U133_Plus_2 excluded due to an elvated NUSE value and RNA degradation. Several other samples are borderline (SM2101400_S23_HG−U133_Plus, M2101404_S25_HG−U133, M2101406_S26_HG−U133_Plus, GSM2101428_S6_HG−U133_Plus), check correlation in GV after import.\"\n",
      "\"Sample GSM211314 was excluded from analysis. The rest of samples were good quality.\"\n",
      "\"Sample GSM2125366 was excluded  due to lower quality.\"\n",
      "\"Sample GSM212558 excluded from analysis, the rest of samples are good quality.\"\n",
      "\"Sample GSM2200529 excluded due to low RLE and lower correlation with some samples. All other samples have a good quality.\"\n",
      "\"Sample GSM2233566 was excluded due to high NUSE and low correlation, all other samples have a good quality.\"\n",
      "\"Sample GSM236912 was excluded from further analysis, the rest of samples were good quality.\"\n",
      "\"Sample GSM241135 was excluded from analysis, the rest of samples were good quality.\"\n",
      "\"Sample GSM24472 was excluded from further analysis. The rest of samples were good quality.\"\n",
      "\"Sample GSM2722755 has been excluded due to lower correlation with other samples. Other samples have a good quality.\"\n",
      "\"Sample GSM272346 was excluded from further analysis. The rest of samples were good quality.\"\n",
      "\"Sample GSM312019 is excluded due to lower expression boxplot, high NUSE, and lower array-array correlation. The rest of samples are ok.\"\n",
      "\"Sample GSM317653 excluded - raw file was corrupted.\"\n",
      "\"Sample GSM324796 has low quality and therefore it is excluded from further analysis.\"\n",
      "\"Sample GSM38346 has an elevated NUSE value, checked correlation after import in GV and is OK. Otherwise all samples are of good quality. Note: samples generated in this study were also re-analyzed together with GSE13896 (HS-00578) under GSE46903 (affymetrix part).\"\n",
      "\"Sample GSM388749 excluded due to low quality. Other samples are OK.\"\n",
      "\"Sample GSM389654 (CFTR genotype D508) excluded from experiment, other samples good QC.\"\n",
      "\"Sample GSM440270 has been excluded due to lower quality. GSE17643 is a subserie of  GSE17672.\"\n",
      "\"Sample GSM458078 is low quality (RNA degradation) and is excluded from further analysis.\"\n",
      "\"Sample GSM463355 excluded for low quality data.\"\n",
      "\"Sample GSM466818 excluded for unusal # of probesets.\"\n",
      "\"Sample GSM53299 was excluded due to low correlation with other samples (and  NUSE >1.05), and sample GSM53308 was excluded due to lower level of probe signal intensity).\\\n",
      "\"Sample GSM584469 was excluded due to unavailable data. Rest of samples-good quality.\"\n",
      "\"Sample GSM622756 was excluded due to high NUSE  (1.1) and lower correlation with the rest of the samples.\"\n",
      "\"Sample GSM652844 was excluded due to lower array−array correlation and higher RLE and NUSE.\"\n",
      "\"Sample GSM654661 was excluded due to an increased RNA degradation and poor correlation.\"\n",
      "\"Sample GSM747332 excluded for RNA degradation.\"\n",
      "\"Sample GSM762920 excluded due to higher NUSE and RNA degradation.\"\n",
      "\"Sample GSM932105 excluded for low quality.\"\n",
      "\"Sample GSM948631 excluded due to low quality data.\"\n",
      "\"Sample GSM960521 was excluded from further analysis, the rest of samples were good quality.\"\n",
      "\"Sample GSM980686 is excluded because of low Correlation and higher Nuse plot. The rest of samples are good quality.\"\n",
      "\"Sample Human_253949444665_S01_GE1_1105_Oct12_2_1.txt (X21) excluded based on RLE and PCA results.\"\n",
      "\"Sample Indotto2HU_U133A_29_06_04 was excluded due to high NUSE and lower array-array correlation.\"\n",
      "\"Sample name issues\"\n",
      "\"Samples 007867_HGU133A.CEL.gz; 007875_HGU133A.CEL.gz and 022440_HGU133A.CEL.gz excluded for low quality data.\"\n",
      "\"Samples AdMinus_3 and AdMinus_4 are excluded from annotation because of bad values in NUSE plot.\"\n",
      "\"Samples annotated under HS-02230.\"\n",
      "\"samples are annotated as subserie of GSE35661 (HS-02091)\"\n",
      "\"Samples are duplicated in MM-001070\"\n",
      "\"Samples are duplicates (subserie) - to HS-01178 study (GSE42488)\"\n",
      "\"Samples are duplicates to HS-01049.\"\n",
      "\"Samples are OK\"\n",
      "\"Samples are OK, except 3 samples with lower correlation- needs to be checked in GV!\"\n",
      "\"Samples are OK. Good correlation. Although in Expression Density Plots- 2 \"populations\" appeared,  in RLE Processed Signal- samples \"GSM2318637\\\n",
      "\"Samples are OK. Superserie GSE52875 consists of subseries GSE52866 (32 samples) and GSE52871 (16 samples), which have been annotated (48 samples). Other  two subseries (GSE52872 and GSE52869) are on the unsupported platform.\"\n",
      "\"Samples are on MM_AFFY_430A_2 platform. All samples have a good quality.\"\n",
      "\"Samples: arrays_251485036302_S01_GE2-v5_95_Feb07_1_1, arrays_251485036302_S01_GE2-v5_95_Feb07_1_2, arrays_251485036302_S01_GE2-v5_95_Feb07_1_3 were excluded, because Cy5 was used for their labeling.\"\n",
      "\"Samples Bonemarrowplasmacells2; Bonemarrowplasmacells3 and Bonemarrowplasmacells4 were excluded due low quality (higher NUSE, Positive Border Elements; and lower correlation).\"\n",
      "\"samples excluded due to duplication:GSM568829, GSM568830, GSM568831, GSM568867, GSM568868, GSM568869\"\n",
      "\"Samples excluded due to high NUSE: GSM1218110, GSM1218114, GSM1218113, GSM1218115, GSM1218111, GSM1218112, GSM1218108, GSM1218109\"\n",
      "\"Samples excluded due to low quality data: GSM734028, GSM734032, GSM734039, GSM734044, GSM734058, GSM734059, GSM734061, GSM734062, GSM734068, GSM734085, GSM734090, GSM734091, GSM734098.\"\n",
      "\"Samples excluded due to quality issue: GSM358114,  GSM358129, GSM358132, GSM358134, GSM358139, GSM358152, GSM358171, GSM358173, GSM358197, GSM358117, GSM358167, GSM358136,  GSM358141, GSM358147 - GSM358149, GSM358165, GSM358170, GSM358186, GSM358203, GSM358205. GSE14323 study contains two different platforms, therefore it was split into two different HS-numbers.\"\n",
      "\"Samples excluded from annotation because of low QC: GSM1360790, GSM1360792, GSM1360802, GSM1360805, GSM1360807, GSM1360809, GSM1360849, GSM1360853. The rest of samples with good QC.\"\n",
      "\"Samples excluded from annotation because of median values were >0.5/< -0,5 in RLE graph or low correlation of samples: GSM1274031, GSM1274035, GSM1274043, GSM1274061, GSM1274083, GSM1274084, GSM1274085, GSM1274086, GSM1274089, GSM1274090, GSM1274091, GSM1274092, GSM1274101, GSM1274102, GSM1274108.\"\n",
      "\"Samples excluded from annotation because they are duplicates with samples which have been already imported in the GV: GSM599911, GSM599912, GSM599913, GSM599914, GSM599915, GSM599916, GSM599917, GSM599918, GSM599919, GSM599920, GSM599921.\"\n",
      "\"Samples exclusion: GSM261086\\\n",
      "\"Samples from platform [HG-U133A] Affymetrix Human Genome U133A Array (GPL96) are annotated under HS-00149. Samples from platform [HG_U95Av2] Affymetrix Human Genome U95 Version 2 Array (GPL8300) are annotated under HS-02410.\"\n",
      "\"Samples GSM1012406,\\\n",
      "\"Samples GSM1047850, GSM1047851, GSM1047852, GSM1047853, GSM1047854, GSM1047855, GSM1047856, GSM1047857, GSM1047858, GSM1047859 (whole group of control healthy skin samples) were excluded due to duplicate detection with samples from HS-01220 (GSE42114). Therefore only 15 samples were annotated under HS-02295.\"\n",
      "\"Samples GSM1098446 and GSM1098453 were excluded due to low quality (high RLE, NUSE and low array-array correlation). 11/13 samples were annotated.\"\n",
      "\"Samples- GSM1126426, GSM1126427, GSM1126428, excluded due to overall lower quality. The other samples are OK.\"\n",
      "\"Samples GSM1161396 and GSM1161401 had low quality on QC plots and are excluded from the study.\"\n",
      "\"Samples GSM1239347 and GSM1239350 excluded due to low correlation (0.7). Other samples have a good quality.\"\n",
      "\"Samples GSM1240925 and GSM1240926 are excluded because of high values in NUSE plot. The rest of samples are OK.\"\n",
      "\"Samples GSM1275077; GSM1275095 and GSM1275097 excluded due to low quality data.\"\n",
      "\"Samples GSM1306168; GSM1306183; GSM1306184; GSM1306191; GSM1306194 and GSM1306198 excluded for low quality.\"\n",
      "\"Samples GSM1321200,\\\n",
      "\"Samples GSM1324621 and GSM1324628 were excluded due to bad QC (higher RLE, higher NUSE, lower array-array correlation).\"\n",
      "\"Samples GSM143568, GSM143569, GSM143570 were excluded from further analysis, the rest of samples were good quality.\"\n",
      "\"Samples GSM1518673, GSM1518674, GSM1518676 and GSM1518679 need to be checked after import. Otherwise samples have a good quality.\"\n",
      "\"Samples GSM1525438_P4 and GSM1525449_P15  excluded due to higher NUSE (≥1,05) and lower correlation with other samples (< 0,75). Quality of other samples is OK.\\\n",
      "\"Samples GSM1532545 and GSM1532546 were excluded due to lower quality.\"\n",
      "\"Samples GSM1534353, GSM1534397, GSM1534398 were excluded due to low correlation.\"\n",
      "\"Samples GSM1541396, GSM1541400 excluded due to lower quality (NUSE above 1.05, different RNA degradation plots).\"\n",
      "\"Samples GSM155932 – GSM155934, GSM155937 – GSM155944, GSM155949, GSM155953, GSM155954, GSM155956 – GSM155959, GSM155963, GSM155985, GSM155986 were excluded, because their duplicates were found in the HS-00437 study. Other samples were of a good quality.\"\n",
      "\"Samples GSM1589520, GSM1589528, GSM1589534,GSM1589554 were excluded from annotation because of low QC. The rest is with good QC.\"\n",
      "\"Samples GSM159919, GSM159921 excluded. All other samples have a good quality.\"\n",
      "\"Samples GSM1615569, GSM1615570, GSM1615571, GSM1615572, GSM1615573, GSM1615574, GSM1615575, GSM1615576 excluded due to low quality, NUSE above 1.15.\"\n",
      "\"Samples GSM1619603 and GSM1619605 were excluded due to high RLE and low array-array correlation.\"\n",
      "\"Samples GSM1647688 and GSM1647689 were excluded from annotation because of low QC. The rest of samples is with good QC.\"\n",
      "\"Samples GSM1659060, GSM1659083, GSM1659103 and GSM1659150 were excluded because they are duplicates with samples from studies HS-02365 and HS-02366. Sample GSM1659100 was excluded because it is duplicate with sample GSM1659088 from this study. Samples GSM1659067, GSM1659092, GSM1659095, GSM1659101, GSM1659123, GSM1659130, GSM1659133, GSM1659134, GSM1659148 were excluded because of low QC (RLE > 0.5 or < -0.5).\"\n",
      "\"Samples GSM1660470, GSM1660490, GSM1660536 have low quality and are excluded from further analysis.\"\n",
      "\"Samples GSM1661652 and GSM1661655 are excluded from the study - low quality in Nuse and Correlation plot.\"\n",
      "\"Samples GSM1833927, GSM1833937, GSM1833938 were excluded from the analysis, based on the NUSE plot values. Rest of samples are OK.\"\n",
      "\"Samples- GSM1842073, GSM1842074, GSM1842075, excluded, because Relative Log Expression is out off ranges (-0.5, 0.5).\"\n",
      "\"Samples GSM18441, GSM18453 excluded, All other samples have a good quality.\"\n",
      "\"Samples GSM1900696, GSM1900697, GSM1900699 excluded due to overall low quality. The other samples are OK.\"\n",
      "\"Samples GSM1975478 and GSM1975480 were excluded due to re-analysis from study GSE71731 (HS-01925).\"\n",
      "\"Samples GSM2107975; GSM2107977 and GSM2107978 excluded due to low correlation with other samples.\"\n",
      "\"Samples GSM2233653, GSM2233659, GSM2233765, GSM2233772, GSM2233807, GSM2233839, GSM2233867, GSM2233910, GSM2234049, GSM2234095, GSM2234103, GSM2234215, GSM2234286, GSM2234309, GSM2234328, GSM2234365, GSM2234387, GSM2234425, GSM2234440, GSM2234468, GSM2234546, GSM2234568 were excluded due to high NUSE and low correlation.\"\n",
      "\"Samples GSM241190, GSM241191, GSM241192 were excluded from further analysis, the rest of samples were good quality.\"\n",
      "\"Samples GSM247593, GSM247594 were excluded from analysis, the rest of samples are good quality.\"\n",
      "\"Samples GSM250849, GSM250853 were excluded from further analysis, the rest of samples were good quality.\"\n",
      "\"Samples GSM267859, GSM267860, GSM267861,GSM267862, GSM267883, GSM267889 were excluded fro further analysis. The rest of samples were good quality.\"\n",
      "\"Samples GSM279222, GSM279240, GSM279263 were excluded from further analysis. The rest of samples are good quality.\"\n",
      "\"Samples GSM3036714, GSM3036715, GSM3036716 excluded due to bad NUSE and correlation plot.\"\n",
      "\"Samples GSM311504, GSM311515, GSM311516, GSM311535, GSM311544, GSM311567, GSM311573, GSM311578, GSM311579 were excluded from further analysis. The rest of samples were good quality.\"\n",
      "\"Samples GSM313041 and GSM313054 were excluded due to lower correlation with other samples.\\\n",
      "\"Samples GSM342374, GSM342375, GSM342458, GSM342327, GSM342329 were excluded due to bad QC.\"\n",
      "\"Samples GSM37491; GSM37492; GSM37493; GSM37494; GSM37495; and GSM37496  were excluded due to higher signal intensity and lower correlation with other samples.\\\n",
      "\"Samples GSM37497 and GSM37498 were excluded from annotation, because of discrepancy in sample descriptions between the publication and the matrix file. It is not possible to describe samples clearly. The rest of samples with good QC.\"\n",
      "\"Samples GSM397336, GSM397366, GSM397367 excluded due to low quality, RLE above 0.5. Double labeling Cy3/Cy5, only Cy3 labeled samples annotated.\"\n",
      "\"Samples GSM4166, GSM4161 excluded. All other samples have a good quality.\"\n",
      "\"Samples GSM418657, GSM418663 were excluded due to high NUSE and low correlation. All other samples have  a good quality.\"\n",
      "\"Samples GSM444361, GSM444362, GSM444363 were excluded due to duplication with MM-00544 (GSE8720). All other samples have a good quality.\"\n",
      "\"Samples GSM455850, GSM455851, GSM455852, GSM455853, GSM455854, GSM455855, GSM455856, GSM455857, GSM455858 are duplicate detected with samples from GSE13159 (HS-00730). Moreover sample GSM455848 has worst QC (borderline). Therefore study GSE18239 is rejected.\"\n",
      "\"Samples GSM497558 and GSM497568 were excluded because of low QC. The rest of samples is OK.\"\n",
      "\"Samples GSM572086 and GSM572089 were excluded due to bad QC (lower expression boxplot, high expression density plot, lower positive border elements and lower array-array correlatioin).\"\n",
      "\"Samples GSM575550; GSM575552 and GSM575186 excluded due to poor QC data.\"\n",
      "\"Samples GSM687660, GSM687730, GSM687695 are excluded due to bad QC (worst RLE, bad NUSE, low array-array correlation and lower correlation matrix in GV).\"\n",
      "\"Samples GSM692518, GSM692519, GSM692520, GSM692521, GSM692522, GSM692523, GSM692524, GSM692525 have low values in NUSE and Correlation plot in comparison to all samples in the study.\"\n",
      "\"Samples GSM787076, GSM787077, GSM787078, GSM787079, GSM787090, GSM787091, GSM787119, GSM787179, GSM787180, GSM787181, GSM787183, GSM787263, GSM787264, GSM787268 are excluded due to high RLE, NUSE and lower array-array correlation.\"\n",
      "\"Samples GSM832065 is excluded due to bad QC (high NUSE, low array-array correlation).\"\n",
      "\"Samples GSM854774 and GSM854775 were excluded due to bad QC (high RLE, high NUSE).\"\n",
      "\"Samples GSM870907, GSM870913 were excluded because of low QC. The rest of samples with good QC.\"\n",
      "\"Samples GSM872328 and GSM872387 excluded due to low quality data.\"\n",
      "\"Samples GSM909435_Pkj_P56_AP103, GSM909454_SB_P56_AP110, GSM909426_Pkj_P28_AP12 were excluded du to high NUSE and intensity.\"\n",
      "\"Samples GSM982971, GSM982972, GSM982973, GSM982974 (whole subserie GSE39985) are excluded due to bad QC (high RLE and NUSE, low array-array correlation).\"\n",
      "\"Samples GSM985655 and GSM985509 excluded for data low quality. Many control samples excluded for duplicity. List of duplicates available upon request.\"\n",
      "\"Samples- GSM998424, GSM998428, GSM998429, GSM998430, GSM998431, GSM998432, GSM998433, GSM998435, GSM998436, excluded due to over all low quality (low expression, RLE and NUSE above ranges, low correlation 0.2). The other samples are OK.\"\n",
      "\"Samples have a good quality.\"\n",
      "\"Samples have a good quality, apart from sample GSM712798, which has been excluded (due to  higher signal intensity and signal variability).\"\n",
      "\"Samples have a good quality, except sample GSM1643662_C_R001 with low correlation 0.76.\"\n",
      "\"Samples have good qc. Sample GSM648481 has lower correlation in comparison with other samples, but the correlation coefficient is within acceptable range.\"\n",
      "\"Samples in matrix file are named differently than .txt files as they appear in QC report\"\n",
      "\"Samples measured in two channels (cy5, cy3) - not supported in GV → experiment rejected\"\n",
      "\"Samples measured in two channels (Cy5/Cy3) - reference is on Cy3 channel, experimental RNA on Cy5 - not supported in GV\"\n",
      "\"Samples Nas484, Nas497 and Nas518 are excluded because of low RNA quality.\"\n",
      "\"!!samples on cy5!!\"\n",
      "\"Samples on cy5 channel can't be processed in our pipeline.\"\n",
      "\"Samples Sample_1 and Sample_4 excluded, RLE above 0.5.\"\n",
      "\"Sample US90703645_252665514002_S01_GE1_1100_Jul11_1_4 excluded for low quality.\"\n",
      "\"Several samples have border or lower correlation. Check in GV.\"\n",
      "\"Signal data have overall very low quality - large spread of RLE values, array-array correlation only 0.1 - 0.2; check of gene expression in GV showed completely random behavior of genes with selections of low/medium/high expressed reference genes → experiment rejected\"\n",
      "\"Signal intensity and sample-to-sample correlation is low for many samples in this study. Having run and checked QC of another 20 studies on this platform, we decided to keep all samples in this study -  consulted with Jana.\"\n",
      "\"Study contains 211 duplicates with samples from HS-02458 (GSE54837). Therefore only 18 samples were annotated under HS-02459.\"\n",
      "\"Study contains two matrix files. First one GSE35661-GPL9454 contains only control samples (already 41patients before training). Second matrix GSE35661-GPL570 contains 24 samples of patients after training. All samples have good QC.\"\n",
      "\"Study GSE12971 contains samples from studies GSE12952 and GSE13459 (plus three extra samples).  The same control samples used in all studies. All samples have a good quality.\"\n",
      "\"Study rejected due to low correlation of the samples (bellow 0.5).\"\n",
      "\"Study rejected due to low quality data from the whole experiment: Expression boxblot less than 6, array-array intensity correlation low (0.3-0.5).\"\n",
      "\"Study rejected due to low quality of the data.\"\n",
      "\"Study rejected due to low quality of the data (2 peaks on the expression density plot of signal data).\"\n",
      "\"Study rejected for low quality data.\"\n",
      "\"Study rejected for low quality data. Positive border elements are in the same range with negative border elements.\"\n",
      "\"sub-optimal QC: GSM595587, GSM595583, GSM595559\"\n",
      "\"suboptimal QC: GSM850527_RSN-11, GSM850535_RSN-4, GSM850536_RSN-5\"\n",
      "\"Subserie annotated as a part of superserie GSE52875 (MM-00773).\"\n",
      "\"SubSerie GSE29846 contains 25 samples which are part of SuperSerie GSE24347 (32 samples). 7/32 samples from SuperSerie GSE24347 are from unsupported Mouse Genome CGH Microarray 244A. Moreover 5 samples from SubSeriGSE29846 (GSM739460, GSM739461, GSM739462, GSM739463, GSM739464) are excluded due to bad QC (lower NUSE, low array-array correlation). Overall 20/32 samples from SuperSerie GSE24347 were annotated.\"\n",
      "\"Subserie GSE34524 was annotated under superserie GSE34525 (HS-02518).\"\n",
      "\"Subserie GSE35118  was annotated under superserie GSE34525 (HS-02518).\"\n",
      "\"Superserie consisting of 2 platforms. Samples analyzed on supported platform are annotated in HS-02037.\"\n",
      "\"Superserie GSE81984 consisting of 2 subseries: GSE81870: 5 samples were annotated under HS-02484. Subserie GSE81982: 1 RNaseq sample was not annotated (HS-02498).\"\n",
      "\"Superserie of 17 subseries on 9 various platforms. Subseries, which are on a supported platforms, are annotated under separate internal GV-IDs.\"\n",
      "\"Superserie of 2 subseries containig 19 samples (on 2 different platforms). Subserie GSE14903 is annotated under RN-00247. Subserie GSE14902 is on “soon available” platform (RN-Affy_320A).\"\n",
      "\"Superserie of GSE34423 (annotated as MM-00670 on MM_AFFY_430_2 platform) and GSE34462 ( NimbleGen MM8 Mouse Promoter plus CpG Island 374K Tiling Array - not supported)\"\n",
      "\"The correct platform is HS_AFFY_U95A (GPL91) - currently not supported in GV\"\n",
      "\"The platform is not yet supported; annotations are available as flat file\"\n",
      "\"The raw data files in .cel format were downloaded from GSE12021. GSE12021 is already in our system (HS-02342 and HS-02343).\"\n",
      "\"There are two populations of samples with big difference in RAW data intensity, Bcl3-/- samples (samples 17-24; log2 median approx. 5) and the rest of samples (samples 1-16; log2 median approx. 8) adverting most probably to technical bias. Samples with high row data intensities (GSM575606, GSM575607, GSM575608, GSM575613, GSM575609, GSM575610, GSM575611, GSM575612) were excluded.\"\n",
      "\"There is a wrong platform label in GEO for this experiment (the experiment is labeled as Human experiment).  QCNM - all samples are OK.\"\n",
      "\"the rest 9 samples were just the same, but on an other chip plattform.\"\n",
      "\"The sample GSM2038282 was excluded because of low QC. The rest of samples is with good QC.\"\n",
      "\"These samples are duplicates of those published  under GEO GSE37448 (MM-00560, Immgen phase 2).\"\n",
      "\"The signal data have been checked by EvaP (a complicated study). Study needs to be check in GV. All samples are OK. GSM776998 will checked in GV.\"\n",
      "\"The study is available as a flat file.\"\n",
      "\"The study is rejected because 7 samples are duplicates with samples from the study HS-02273. The study HS-02273 was annotated because contains other samples.\"\n",
      "\"The study is rejected, because samples are duplicates with samples from superserie GSE60301. The samples are annotated under HS-02350.\"\n",
      "\"The study rejected for serious inconsistencies in experiment description.\"\n",
      "\"The study was removed because it is duplicate of the study HS-01085.\"\n",
      "\"The study will not be annotated because it contains 105 duplicates from studies HS-02364 and HS-02365. Information from the matrix file (gender, sex) was matched with samples from the study HS-02365, which will be used for annotation.\"\n",
      "\"The subserie 68364 is a part of the superserie GSE68387. All data have a good quality.\"\n",
      "\"The subserie GSE29617 is a part  of superserie GSE29619. Study rejected for low quality data. Positive border elements are in the same range with negative border elements.\"\n",
      "\"The subserie GSE60355 is a part of superserie GSE60356. All samples have a good quality.\"\n",
      "\"The subserie GSE60684 is a part of superseries GSE68387 and GSE60693. All data have a good quality.\"\n",
      "\"The subserie GSE68120 is a part of the superserie GSE68387. All data have a good quality.\"\n",
      "\"The subserie GSE68121 is a part of the superserie GSE68387. 19 samples excluded for RNA degradation: GSM1664123, GSM1664083, GSM1664094, GSM1664099, GSM1664103, GSM1664119, GSM1664085, GSM1664088, GSM1664097, GSM1664120, GSM1664077, GSM1664078, GSM1664079, GSM1664100, GSM1664101, GSM1664102, GSM1664116, GSM1664117, GSM1664118.\"\n",
      "\"The subserie GSE80018 is a part of the superserie GSE68387. All samples have a good quality.\"\n",
      "\"The superserie GSE16768 consist of three subseries: GSE16656, GSE16766, GSE16767. Samples from all of them are included. Sample GSM420252 excluded for weak positive border elements signal.\"\n",
      "\"Third party re-analysis of samples originally analysed elsewhere. Samples can be found under the original study numbers: HS-00037, HS-00038, HS-00511, HS-00794, HS-00853, HS-00870, HS-00913, HS-00919, HS-00930, HS-01074, HS-01075, HS-01076, HS-01077, HS-01080, HS-01143, HS-01307, HS-02089, HS-02090, HS-02091, HS-02092, HS-02095, HS-02137, HS-02138, HS-02264, HS-02335, HS-02336, HS-02337, HS-02338, HS-02339.\"\n",
      "\"Third party re-analysis of samples originally analysed elsewhere. Samples can be found under the original study numbers: HS-00074, HS-00075, HS-00108, HS-00149, HS-00184, HS-00192, HS-00216, HS-00342, HS-00353, HS-00872, HS-00879, HS-00895, HS-02140, HS-02149, HS-02361 and HS-02362.\"\n",
      "\"Third party re-analysis. Samples are analyzed under RN-00198.\"\n",
      "\"third-party re-analysis study\"\n",
      "\"This experiment is on currently unsupported platform. The subserie GSE68111 is a part of the superserie GSE68387.\"\n",
      "\"This is duplicated entry, the same GSE number is also assigned to HS-01720\"\n",
      "\"This study is a part of superserie GSE56768. Superserie GSE56768 is composed of subserie GSE56766 and GSE56767, both on the same platform. Both subseries annotated together in the study with HS number HS-02178. QC good.\"\n",
      "\"This study is a part of superserie GSE56768. Superserie GSE56768 is composed of subseries GSE56766 and GSE56767, both on the same platform. Both subseries were annotated together in the study with HS number HS-02178. good qc\"\n",
      "\"Three samples (GSM1282095, GSM1585329, GSM2112445) excluded due to elevated NUSE values and low correlation to other samples. Several other samples borderline but kept.\"\n",
      "\"Three samples (GSM605790,  GSM791118, GSM791127) excluded due to highly elevated NUSE values and very low correlation with the other samples. Several other samples are of borderline quality (NUSE, correlation) but decided to keep them.\"\n",
      "\"Two Agilent platforms, QC needs to be consulted\"\n",
      "\"two channels!!!!\"\n",
      "\"Type 1 diabetes samples are part of HS-01049, already public\"\n",
      "\"unclear annotation in repository = not annotated: twin5, twin6, twin7, twin8\"\n",
      "\"unclear description = not annotated: MolOBB_Plate2__C9.CEL, MolOBB_Plate2__C11.CEL, MolOBB_Plate2_D2.CEL, MolOBB_Plate_1_B10.CEL, MolOBB_Plate_1_D12a.CEL, MolOBB_Plate_1_B8.CEL, MolOBB_Plate1_A3.CEL, MolOBB_Plate_3_C2.CEL, MolOBB_Plate2_H1.CEL, MolOBB_Plate2_H12.CEL, MolOBB_Plate2_G8.CEL, MolOBB_Plate2_F4.CEL; bad QC: MolOBB__Plate2__B7.CEL.gz, MolOBB__Plate2__B2.CEL.gz, MolOBB__Plate2__B1.CEL.gz, MolOBB__Plate2__B5.CEL.gz, MolOBB__Plate2__B9.CEL.gz, MolOBB__Plate2__B8.CEL.gz, MolOBB__Plate2__B11.CEL.gz, MolOBB__Plate2__B10.CEL.gz, MolOBB__Plate2__B3.CEL.gz, MolOBB__Plate2__B4.CEL.gz, MolOBB_Plate2__C1.CEL.gz, MolOBB__Plate2__B6.CEL.gz, MolOBB__Plate2__B12.CEL.gz, MolOBB_Plate2_D12.CEL.gz, MolOBB_Plate_1_D3.CEL.gz, MolOBB_Plate_1_D11a.CEL.gz, MolOBB_Plate_1_C9.CEL.gz, MolOBB_Plate_1_D1.CEL.gz, MolOBB_Plate_1_F9.CEL.gz\"\n",
      "\"unclear description = not annotated: MolTWIN_OX0284_1E5.CEL, MolTWIN_OX0183_2G7.CEL\"\n",
      "\"un-published, already in a database as HS-00687\"\n",
      "\"untreated controls samples were removed due to duplication with HS-01322\"\n",
      "\"unusual format of RAW data which is not compatible with the internal pipeline\"\n",
      "\"Waiting for QC results (Martin Holub).\"\n",
      "\"Was added as a part of HS-01040\"\n",
      "\"Whole experiment has low array-array correlation (0.7); 10 samples excluded due to low tech. quality ( high NUSE and  correlation <0.7): GSM428924, GSM428940, GSM428946, GSM429412, GSM429422; GSM431145, GSM431146, GSM431155, GSM431157, GSM431158\"\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "awk -F\"\\t\" -v OFS=\"\\t\" 'tolower($3) !~ /rna/ {print $4}' exp_qc_notes.tsv | sort | uniq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Save the strings and filter them out manually."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "awk -F\"\\t\" -v OFS=\"\\t\" 'tolower($3) !~ /rna/ {print $4}'  exp_qc_notes.tsv| sort | uniq | grep -P -i \"(bad qc|low qc|low|nuse|bad quality|low quality|lower quality|rle|correlation|corelation|excluded|clustering|degradation)\" > qcnotes_badqc_patterns.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<!-- Not done\n",
    "- Next, get expereriments that were rejected with all samples, ignore the ones that have duplicate in their annotation\n",
    "`awk -F\"\\t\" -v OFS=\"\\t\" 'tolower($3) !~ /rna/ && $2 == 1 {print $1,$3,$4}' exp_qc_notes.tsv | grep -v -i -P \"(duplicate|dual-labeled|cy5)\" | cut -f1,2 > bad_exps.txt`\n",
    "-->\n",
    "\n",
    "Next, get expereriments that were rejected with all samples, take only those that match your pattern.\n",
    "\n",
    "This is still error prone as experiment could have been fully rejected and have only some samples with supposingly bad QC (e.g HS-01353)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "awk -F\"\\t\" -v OFS=\"\\t\" 'tolower($3) !~ /rna/ && $2 == 1 {print $1,$2,$3,$4}' exp_qc_notes.tsv | grep -Fwf qcnotes_badqc_patterns.txt | cut -f1,3 | \n",
    "grep -P -v \"(HS-01353|HS-00391|HS-01418|HS-02051|HS-02411)\" > bad_exps.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next get experiments and their QC notes that were rejected only partially. It is possible to manually extract some sample names that were deemed to have bad QC from here. This data is probably the most usable. \n",
    "\n",
    "Additonally, it is also this list where experiments that have been flagged as QCNM_Issue, but not REJECTED (although they have usually all samples flaged as bad QC) live. One can fish them manually from this file, but they probably should not be used for labeling (similarly as full REJECTED experiments do not give stellar results.). You will find them in `paritaly_bad_exps_manual_annotation.tsv` denoted by `*` (ie. all samples bad)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "awk -F\"\\t\" -v OFS=\"\\t\" 'tolower($3) !~ /rna/ && $2 == 0 {print $1,$2,$3,$4}' exp_qc_notes.tsv | grep -Fwf qcnotes_badqc_patterns.txt | cut -f1,3,4 > partially_bad_exps.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, get samples that come from experiments wehere we are fairly sure that the rejection has happened due to bad qc. There is 1651 of them. The platform of origin can be fetched from `bad_exps.txt`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat bad_exps.txt | cut -f1 | grep -Fwf - perhaps_badqc_samples.txt > bad_samples.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next look if something fruitful can be fetched from comments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cut -f2 -d\":\" exp_comments_norm.json | sed -e 's/^ \"//' -e 's/\",$//' | grep -v -P \"(^{|}$)\" | grep -P -i \"(bad qc|low qc|low|nuse|bad quality|low quality|lower quality|rle|correlation|corelation|excluded|clustering|degradation)\" | sort | uniq > bad_comment_patterns.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, you would have to go through this list and see if you can fetch something meaningful from it manually. Not done."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat bad_comment_patterns.txt | grep -Fwf - exp_comments_norm.json | cut -f1 -d\":\" | sed -e 's/^\\s\\+\"//' -e 's/\"$//' > exps_with_bad_comments.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "v1=$(awk -F\"\\t\" -v OFS=\"\\t\" '{print $0, \"1\"}' processed_and_imported_samples.txt)\n",
    "v2=$(awk -F\"\\t\" -v OFS=\"\\t\" '{print $0, \"0\"}' bad_samples.txt)\n",
    "echo \"$v2\" \"$v1\" | sort | uniq > training_labels.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This seems legit, but probably will have to discard cases when sample called \"1\" or similar.\n",
    "\n",
    "Confirmation:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "comm -12 <(sort processed_and_imported_samples.txt) <(sort bad_samples.txt)\n",
    "# is empty vector"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Getting experiments to process"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cut -f1 processed_and_imported_samples.txt | cat <(cut -f1 bad_samples.txt) - | sort | uniq > exps_to_process.txt\n",
    "\n",
    "grep -Fwf  exps_to_process.txt exp_qc_notes.tsv | cut -f1,3 > exps_to_process_with_platform.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "or to get it separately with some filtering:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cut -f1 bad_samples.txt | sort | uniq |  grep -Fwf -  exp_qc_notes.tsv | cut -f1,3 |  grep \"^HS\" | grep \"AFFY\" | grep -v \"MIRNA\" >  bad_exps_to_process_with_platform.tsv\n",
    "# 18 of them\n",
    "\n",
    "cut -f1 processed_and_imported_samples.txt | sort | uniq |  grep -Fwf -  exp_qc_notes.tsv | cut -f1,3 |  grep \"^HS\" | grep \"AFFY\" | grep -v \"MIRNA\" >  good_exps_to_process_with_platform.tsv\n",
    "# 1512 of them"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here take only fraction from the much larger group:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat <(shuf -n 100 good_exps_to_process_with_platform.tsv) bad_exps_to_process_with_platform.tsv > exps_to_process_final.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Conclusion:\n",
    "\n",
    "labels is a vector of \\[processed_and_imported_samples.txt, bad_samples.txt\\]. Whenever we get a sample that is in this vector, we know what label to assing to it. How many samples we can get with such assignment depends on how many experiments we will process. You can get list of these experiments as (you can narrow them down to platform later):"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting the ground truth, revisited:\n",
    "\n",
    "Ok, so far the results have not been stellar. One option is trying more powerful model, but even more fruitful could be getting better data. This is what I will try now:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mholub/tmp/mlQC/labels\n"
     ]
    }
   ],
   "source": [
    "%cd ~/tmp/mlQC/labels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat all_good_samples.tsv all_gv_samples.tsv | sort | uniq | \n",
    "grep -F <(cut -f1 paritaly_bad_exps_manual_annotation.tsv | tr -d '\"') > paritaly_bad_exps_samples.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# cat <(cat paritaly_bad_exps_manual_annotation.tsv | tr -d '\"' | \n",
    "# grep -F - paritaly_bad_exps_samples.tsv |  \n",
    "# awk -F\"\\t\" -v OFS=\"\\t\" '{print $0,\"0\"}') <(cat paritaly_bad_exps_manual_annotation.tsv | tr -d '\"' | \n",
    "# grep -v -F - paritaly_bad_exps_samples.tsv |  \n",
    "# awk -F\"\\t\" -v OFS=\"\\t\" '{print $0,\"1\"}') > paritaly_bad_exps_samples_final.tsv\n",
    "\n",
    "cat <(cut -f2 paritaly_bad_exps_manual_annotation.tsv | tr -d '\"' | grep -v \"\\*\"| \n",
    "grep -f - paritaly_bad_exps_samples.tsv |  \n",
    "awk -F\"\\t\" -v OFS=\"\\t\" '{print $0,\"0\"}') <(cut -f2 paritaly_bad_exps_manual_annotation.tsv | tr -d '\"' | grep -v \"\\*\"| \n",
    "grep -v -f - paritaly_bad_exps_samples.tsv |  \n",
    "awk -F\"\\t\" -v OFS=\"\\t\" '{print $0,\"1\"}') > paritaly_bad_exps_samples_final.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get the experiments to be processed:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cut -f1 paritaly_bad_exps_samples_final.tsv | sort | uniq |\n",
    "awk -F\"\\t\" -v OFS=\" \" '{print \"NEB_P1\",$1,\"affy\"}' > paritaly_bad_exps_to_process_final_submitqcn.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "## Appendix"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Alternative approaches to pull stuff from MongoDB\n",
    "(see section [Experiment Comments](#Experiment-Comments))\n",
    "\n",
    "```SQL\n",
    "/* does no_ work\n",
    "var mapFunction1 = function(){\n",
    "  emit(this.nbr, this.comments.comment);\n",
    "};\n",
    "\n",
    "var reduceFunction1 = function(key, values) {\n",
    "  return values ;\n",
    "};\n",
    "\n",
    "db.experiments.mapReduce(\n",
    "  mapFunction1,\n",
    "  reduceFunction1,\n",
    "  {\n",
    "    out : {inline : 1}\n",
    "  }\n",
    ")\n",
    "*/\n",
    "\n",
    "/* works, but the arrays are not completely concatenated */\n",
    "db.experiments.aggregate([\n",
    "  {_id: \"$nbr\", $project: {comments: \"$comments.comment\"}},\n",
    "  {$out: \"mhresults\"}\n",
    "])\n",
    "/* same output as above\n",
    "db.experiments.aggregate([\n",
    "  {$project:\n",
    "    {_id: \"$nbr\",\n",
    "    comments:\n",
    "    { $map: {\n",
    "      input: \"$comments.comment\",\n",
    "      as: \"comments\",\n",
    "      in : {$setUnion: \"$$comments\"}}}}}\n",
    "  ])\n",
    " */\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Fixing labels that were badly adjusted\n",
    "\n",
    "In development, I have erroneously deleted \"cel\" from inside of a few sample names. Not to have to reprocess the data with fixed name generation, you can use SequenceMatcher to match with expected missing strings. This is not used n the pipeline anymore as is is slow."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "import difflib\n",
    "\n",
    "aa=\"HS-02133_GSM985442-D-DMO-72-R-1-1R__HG-U133_Plus_2_\"\n",
    "bb=\"HS-02133_GSM985442_CEL-D-DMO-72-R-1-1R__HG-U133_Plus_2_\"\n",
    "s = difflib.SequenceMatcher(None, aa, bb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "_CEL\n",
      "s_2_\n"
     ]
    }
   ],
   "source": [
    "str_size = 0\n",
    "for block in s.get_matching_blocks():\n",
    "    str_dif = None\n",
    "    sig_ndif = block.a - block.b\n",
    "    if sig_ndif < 0:\n",
    "        str_dif = b[str_size:str_size+abs(sig_ndif)]\n",
    "    elif sig_ndif > 0:\n",
    "        str_dif = a[str_size:str_size+abs(sig_ndif)]\n",
    "    if str_dif: print(str_dif)\n",
    "    str_size += block.size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yay\n"
     ]
    }
   ],
   "source": [
    "a = \"HS-02133_GSM985509-009-TOL1-72-R-1-2__HG-U133_Plus_2_\"\n",
    "b = \"HS-02133_GSM985509_CEL-009-TOL1-72-R-1-2__HG-U133_Plus_2_\"\n",
    "str_diff = \"\"\n",
    "for d in difflib.ndiff(a,b):\n",
    "    if d.startswith((\"+\",\"-\")):str_diff += d[2]\n",
    "\n",
    "import re\n",
    "if re.search(\"(cel|gz|zip|txt|gpr)\", str_diff, re.IGNORECASE):\n",
    "    print(\"yay\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is how you can apply it.\n",
    "```python\n",
    "with open(self.flabels, 'r') as rf:\n",
    "            rf = csv.reader(rf, delimiter = delimiter)\n",
    "    for i, row in enumerate(rf):\n",
    "        sname = \"_\".join(row[0:2]) # prefixed name\n",
    "\n",
    "        # Check if this label is in set of our samples.\n",
    "        sname_check = \"_\".join(row[0:2]) if is_prefixed else row[1]\n",
    "        if sname_check not in samples_set: continue\n",
    "\n",
    "            # This solves namin incesistency but is slow\n",
    "            # TODO: Remove once naming corrrect\n",
    "            match = difflib.get_close_matches(sname, self.samples, n = 1, cutoff = 0.6)\n",
    "            if match:\n",
    "                str_diff = \"\"\n",
    "                for d in difflib.ndiff(sname,match[0]):\n",
    "                    if d.startswith((\"+\", \"-\")): str_diff += d[2]\n",
    "                if re.search(\"(cel|gz|zip|txt|gpr)\", str_diff, re.IGNORECASE):\n",
    "                    sname = match[0]\n",
    "            else:\n",
    "                continue # we do not care about this sample\n",
    "```"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": false,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": true,
   "toc_position": {
    "height": "334px",
    "left": "738px",
    "top": "111px",
    "width": "272px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
